diff --git a/common/libretro.c b/common/libretro.c
index 4f48b90..8d28885 100644
--- a/common/libretro.c
+++ b/common/libretro.c
@@ -572,7 +572,7 @@ void retro_get_system_av_info(struct retro_system_av_info *info)
    info->geometry.base_height  = height;
    info->geometry.max_width    = width;
    info->geometry.max_height   = height;
-   info->geometry.aspect_ratio = 4.0 / 3.0;
+   info->geometry.aspect_ratio = 3.0 / 2.0;
 }

 void retro_set_environment(retro_environment_t cb)
diff --git a/common/libretro_core_options.h b/common/libretro_core_options.h
index e15b3a3..8eff659 100644
--- a/common/libretro_core_options.h
+++ b/common/libretro_core_options.h
@@ -42,6 +42,7 @@ struct retro_core_option_definition option_defs_us[] = {
          { "400x224",   NULL },
          { "400x240",   NULL },
          { "480x272",   NULL },
+         { "480x320",   NULL },
          { "512x224",   NULL },
          { "512x240",   NULL },
          { "512x384",   NULL },
@@ -71,7 +72,7 @@ struct retro_core_option_definition option_defs_us[] = {
 #elif defined(DINGUX)
       "320x240"
 #else
-      "320x200"
+      "480x320"
 #endif
    },
    {
@@ -145,7 +146,7 @@ struct retro_core_option_definition option_defs_us[] = {
          { "enabled",   "Enabled" },
          { NULL, NULL },
       },
-      "disabled"
+      "enabled"
    },
    {
       "tyrquake_analog_deadzone",
