diff --git a/Core/System.cpp b/Core/System.cpp
index 6a8f3ab1f0..0dcb137e1e 100644
--- a/Core/System.cpp
+++ b/Core/System.cpp
@@ -104,6 +104,12 @@ FileLoader *PSP_LoadedFile() {
 	return g_loadedFile;
 }
 
+#ifdef __ANDROID__
+static bool isAndroid = true;
+#else
+static bool isAndroid = false;
+#endif
+
 void ResetUIState() {
 	globalUIState = UISTATE_MENU;
 }
@@ -724,7 +730,7 @@ void PSP_RunLoopFor(int cycles) {
 Path GetSysDirectory(PSPDirectories directoryType) {
 	const Path &memStickDirectory = g_Config.memStickDirectory;
 	Path pspDirectory;
-	if (!strcasecmp(memStickDirectory.GetFilename().c_str(), "PSP")) {
+	if (isAndroid && (!strcasecmp(memStickDirectory.GetFilename().c_str(), "PSP"))) {
 		// Let's strip this off, to easily allow choosing a root directory named "PSP" on Android.
 		pspDirectory = memStickDirectory;
 	} else {
diff --git a/libretro/libretro.cpp b/libretro/libretro.cpp
index 222cc98915..7dacda1af5 100644
--- a/libretro/libretro.cpp
+++ b/libretro/libretro.cpp
@@ -1231,19 +1231,13 @@ void retro_init(void)
 
    Path retro_base_dir;
    Path retro_save_dir;
-   const char* dir_ptr = NULL;
-   if (environ_cb(RETRO_ENVIRONMENT_GET_SYSTEM_DIRECTORY, &dir_ptr) && dir_ptr)
-      retro_base_dir = Path(dir_ptr);
-
-   if (environ_cb(RETRO_ENVIRONMENT_GET_SAVE_DIRECTORY, &dir_ptr) && dir_ptr)
-      retro_save_dir = Path(dir_ptr);
-
-   retro_base_dir /= "PPSSPP";
+   retro_base_dir /= "/roms/gamedata/ppsspp";
+   retro_save_dir /= "/roms/gamedata/ppsspp";
 
    g_Config.currentDirectory = retro_base_dir;
    g_Config.defaultCurrentDirectory = retro_base_dir;
    g_Config.memStickDirectory = retro_save_dir;
-   g_Config.flash0Directory = retro_base_dir / "flash0";
+   g_Config.flash0Directory = retro_base_dir / "assets/flash0";
    g_Config.internalDataDirectory = retro_base_dir;
    g_Config.bEnableNetworkChat = false;
    g_Config.bDiscordRichPresence = false;
