diff --git a/Core/Util/PathUtil.cpp b/Core/Util/PathUtil.cpp
index 96be0eac06..2e43952964 100644
--- a/Core/Util/PathUtil.cpp
+++ b/Core/Util/PathUtil.cpp
@@ -10,6 +10,12 @@
 #include "Core/Config.h"
 #include "Common/VR/PPSSPPVR.h"
 
+#ifdef __ANDROID__
+static bool isAndroid = true;
+#else
+static bool isAndroid = false;
+#endif
+
 Path FindConfigFile(const Path &searchPath, std::string_view baseFilename, bool *exists) {
 	// Don't search for an absolute path.
 	if (baseFilename.size() > 1 && baseFilename[0] == '/') {
@@ -45,7 +51,7 @@ Path FindConfigFile(const Path &searchPath, std::string_view baseFilename, bool
 Path GetSysDirectory(PSPDirectories directoryType) {
 	const Path &memStickDirectory = g_Config.memStickDirectory;
 	Path pspDirectory;
-	if (!strcasecmp(memStickDirectory.GetFilename().c_str(), "PSP")) {
+	if (isAndroid && (!strcasecmp(memStickDirectory.GetFilename().c_str(), "PSP"))) {
 		// Let's strip this off, to easily allow choosing a root directory named "PSP" on Android.
 		pspDirectory = memStickDirectory;
 	} else {
diff --git a/libretro/libretro.cpp b/libretro/libretro.cpp
index 3b2f10317e..0b3cf6b69c 100644
--- a/libretro/libretro.cpp
+++ b/libretro/libretro.cpp
@@ -1240,14 +1240,8 @@ void retro_init(void)
 
    Path retro_base_dir;
    Path retro_save_dir;
-   const char* dir_ptr = NULL;
-   if (environ_cb(RETRO_ENVIRONMENT_GET_SYSTEM_DIRECTORY, &dir_ptr) && dir_ptr)
-      retro_base_dir = Path(dir_ptr);
-
-   if (environ_cb(RETRO_ENVIRONMENT_GET_SAVE_DIRECTORY, &dir_ptr) && dir_ptr)
-      retro_save_dir = Path(dir_ptr);
-
-   retro_base_dir /= "PPSSPP";
+   retro_base_dir /= "/roms/gamedata/ppsspp";
+   retro_save_dir /= "/roms/gamedata/ppsspp";
 
    // Check if '<system_dir>/PPSSPP/compat.ini' exists, if not we can assume
    // the user is missing the assets entirely, so let's warn them about it.
@@ -1288,7 +1282,7 @@ void retro_init(void)
    g_Config.currentDirectory = retro_base_dir;
    g_Config.defaultCurrentDirectory = retro_base_dir;
    g_Config.memStickDirectory = retro_save_dir;
-   g_Config.flash0Directory = retro_base_dir / "flash0";
+   g_Config.flash0Directory = retro_base_dir / "assets/flash0";
    g_Config.internalDataDirectory = retro_base_dir;
    g_Config.bEnableNetworkChat = false;
    g_Config.bDiscordRichPresence = false;
