diff --git a/libretro/libretro.cpp b/libretro/libretro.cpp
index 9b4a20bd9..2ae9c173f 100644
--- a/libretro/libretro.cpp
+++ b/libretro/libretro.cpp
@@ -1000,6 +1000,7 @@ void retro_init(void)
    g_Config.internalDataDirectory = retro_base_dir;
    g_Config.bEnableNetworkChat = false;
    g_Config.bDiscordPresence = false;
+   g_Config.iMemStickSizeGB = 16;
 
    VFSRegister("", new DirectoryAssetReader(retro_base_dir));
 
