diff --git a/platform/libretro/libretrohosthelpers.h b/platform/libretro/libretrohosthelpers.h
index e898bf3..64879e1 100644
--- a/platform/libretro/libretrohosthelpers.h
+++ b/platform/libretro/libretrohosthelpers.h
@@ -1,4 +1,5 @@
 #include <string.h>
+#include <cstdint>
 
 void setInputState(
     uint8_t kDown,
diff --git a/source/PicoRam.h b/source/PicoRam.h
index ff4c4e9..6549f15 100644
--- a/source/PicoRam.h
+++ b/source/PicoRam.h
@@ -1,6 +1,7 @@
 #pragma once
 
-#include <cstring> 
+#include <cstring>
+#include <cstdint> 
 #include <string>
 
 /*
diff --git a/source/emojiconversion.cpp b/source/emojiconversion.cpp
index 2db1635..c8698d6 100644
--- a/source/emojiconversion.cpp
+++ b/source/emojiconversion.cpp
@@ -21,6 +21,7 @@
 #include <string>
 #include <codecvt>
 #include <cstring>
+#include <cstdint>
 #include <regex>
 #include <string_view>
 #include <map>
diff --git a/source/fontdata.cpp b/source/fontdata.cpp
index 1fdda14..bfba0ec 100644
--- a/source/fontdata.cpp
+++ b/source/fontdata.cpp
@@ -1,6 +1,7 @@
 
 //taken from tac08: https://github.com/0xcafed00d/tac08
 #include <string>
+#include <cstdint>
 
 const char* font_ss_str = R"(77700000000000000000000000000000000000000000000000700000700000007770000000000000707000000000000000000000000000007070000007000000
 77700000777000007770000070700000707000007070000007700000770000007000000000700000777000000000000000000000000000007070000070700000
diff --git a/source/stringToDataHelpers.cpp b/source/stringToDataHelpers.cpp
index 36008d5..83505c2 100644
--- a/source/stringToDataHelpers.cpp
+++ b/source/stringToDataHelpers.cpp
@@ -1,4 +1,5 @@
 #include <string>
+#include <cstdint>
 
 #include "stringToDataHelpers.h"
 
