diff --git a/sound/soc/codecs/es8316.c b/sound/soc/codecs/es8316.c
index b303ebbd5..d9e33214a 100644
--- a/sound/soc/codecs/es8316.c
+++ b/sound/soc/codecs/es8316.c
@@ -66,6 +66,7 @@
 	{0x50, 0x00}, {0x51, 0x00}, {0x52, 0x00}, {0x53, 0x00},
 };
 
+#define ES8316_CONFIG_REGISTERS 0x50
 /* codec private data */
 struct es8316_priv {
 	struct regmap *regmap;
@@ -84,6 +85,7 @@
 	bool spk_active_level;
 
 	int pwr_count;
+	uint8_t config_save[ES8316_CONFIG_REGISTERS];
 };
 
 /*
@@ -957,39 +959,39 @@
 	return 0;
 }
 
+static void save_es8316_config(struct snd_soc_codec *codec, struct es8316_priv *es8316)
+{
+	int i;
+
+	for (i = 0; i < ES8316_CONFIG_REGISTERS; i++) {
+		es8316->config_save[i] = (uint8_t)snd_soc_read(codec, i);
+	}
+}
+
+static void restore_es8316_config(struct snd_soc_codec *codec, struct es8316_priv *es8316)
+{
+	int i;
+
+	for (i = 0; i < ES8316_CONFIG_REGISTERS; i++) {
+		snd_soc_write(codec, i, es8316->config_save[i]);
+	}
+}
+
 static int es8316_suspend(struct snd_soc_codec *codec)
 {
+	struct es8316_priv *es8316 = snd_soc_codec_get_drvdata(codec);
+
+	save_es8316_config(codec, es8316);
+
 	return 0;
 }
 
 static int es8316_resume(struct snd_soc_codec *codec)
 {
 	struct es8316_priv *es8316 = snd_soc_codec_get_drvdata(codec);
-	int ret;
 
-	es8316_reset(codec); /* UPDATED BY DAVID,15-3-5 */
-	ret = snd_soc_read(codec, ES8316_CLKMGR_ADCDIV2_REG05);
-	if (!ret) {
-		es8316_init_regs(codec);
-		snd_soc_write(codec, ES8316_GPIO_SEL_REG4D, 0x00);
-		/* max debance time, enable interrupt, low active */
-		snd_soc_write(codec, ES8316_GPIO_DEBUNCE_INT_REG4E, 0xf3);
-		/* es8316_set_bias_level(codec, SND_SOC_BIAS_OFF); */
-		snd_soc_write(codec, ES8316_CPHP_OUTEN_REG17, 0x00);
-		snd_soc_write(codec, ES8316_DAC_PDN_REG2F, 0x11);
-		snd_soc_write(codec, ES8316_CPHP_LDOCTL_REG1B, 0x03);
-		snd_soc_write(codec, ES8316_CPHP_PDN2_REG1A, 0x22);
-		snd_soc_write(codec, ES8316_CPHP_PDN1_REG19, 0x06);
-		snd_soc_write(codec, ES8316_HPMIX_SWITCH_REG14, 0x00);
-		snd_soc_write(codec, ES8316_HPMIX_PDN_REG15, 0x33);
-		snd_soc_write(codec, ES8316_HPMIX_VOL_REG16, 0x00);
-		if (!es8316->hp_inserted)
-			snd_soc_write(codec, ES8316_SYS_PDN_REG0D, 0x3F);
-		snd_soc_write(codec, ES8316_SYS_LP1_REG0E, 0xFF);
-		snd_soc_write(codec, ES8316_SYS_LP2_REG0F, 0xFF);
-		snd_soc_write(codec, ES8316_CLKMGR_CLKSW_REG01, 0xF3);
-		snd_soc_write(codec, ES8316_ADC_PDN_LINSEL_REG22, 0xc0);
-	}
+	restore_es8316_config(codec, es8316);
+
 	return 0;
 }
 
