diff --git a/deps/libgo2/src/display.c b/deps/libgo2/src/display.c
index b91cf0624b..da433890e5 100644
--- a/deps/libgo2/src/display.c
+++ b/deps/libgo2/src/display.c
@@ -597,15 +597,15 @@ void go2_surface_blit(go2_surface_t* srcSurface, int srcX, int srcY, int srcWidt
             break;

         case GO2_ROTATION_DEGREES_90:
-            src.rotation = HAL_TRANSFORM_ROT_90;
+            src.rotation = 0;
             break;

         case GO2_ROTATION_DEGREES_180:
-            src.rotation = HAL_TRANSFORM_ROT_180;
+            src.rotation = 0;
             break;

         case GO2_ROTATION_DEGREES_270:
-            src.rotation = HAL_TRANSFORM_ROT_270;
+            src.rotation = 0;
             break;

         default:
