diff --git a/gfx/gfx_widgets.c b/gfx/gfx_widgets.c
index 29e956763e..3caec1f48d 100644
--- a/gfx/gfx_widgets.c
+++ b/gfx/gfx_widgets.c
@@ -931,6 +931,9 @@ void gfx_widgets_iterate(
             p_disp,
             settings, width, height, fullscreen, true);
 
+      if (p_disp->menu_driver_id == MENU_DRIVER_ID_OZONE)
+         scale_factor = scale_factor * 1.500f;
+
    /* Check whether screen dimensions or menu scale
     * factor have changed */
    if ((scale_factor != p_dispwidget->last_scale_factor) ||
