diff -rupN retroarch.orig/gfx/drivers/oga_gfx.c retroarch.new/gfx/drivers/oga_gfx.c
--- retroarch.orig/gfx/drivers/oga_gfx.c	2021-01-03 07:37:05.039434346 -0500
+++ retroarch.new/gfx/drivers/oga_gfx.c	2021-01-03 07:37:35.723731399 -0500
@@ -629,9 +629,17 @@ static bool oga_gfx_frame(void *data, co
 
    if (msg)
    {
+      settings_t *settings                 = config_get_ptr();
+
+      if (settings->bools.video_oga_vertical_enable) {
+      oga_blit(vid->msg_surface, 0, 0, vid->msg_width, vid->msg_height,
+            page_surface, 0, 0, vid->msg_width, vid->msg_height,
+            0, vid->scale_mode, 0xff0105);
+      } else {
       oga_blit(vid->msg_surface, 0, 0, vid->msg_width, vid->msg_height,
             page_surface, 0, 0, vid->msg_height, vid->msg_width,
             HAL_TRANSFORM_ROT_270, vid->scale_mode, 0xff0105);
+      }
    }
 
    if (unlikely(drmModeSetCrtc(vid->fd, vid->crtc_id, page->fb_id, 0, 0, &vid->connector_id, 1, &vid->mode) != 0))
@@ -728,6 +736,11 @@ static void oga_set_rotation(void *data,
    if (!vid)
       return;
 
+   settings_t *settings                 = config_get_ptr();
+   
+   if (settings->bools.video_oga_vertical_enable)
+   rotation = 0;
+
    switch (rotation)
    {
    case 0:
diff -rupN retroarch.orig/gfx/drivers/oga_gfx.c.rej retroarch.new/gfx/drivers/oga_gfx.c.rej
--- retroarch.orig/gfx/drivers/oga_gfx.c.rej	1969-12-31 19:00:00.000000000 -0500
+++ retroarch.new/gfx/drivers/oga_gfx.c.rej	2021-01-03 07:37:35.723731399 -0500
@@ -0,0 +1,15 @@
+--- gfx/drivers/oga_gfx.c	2020-11-15 16:16:12.678114070 -0500
++++ gfx/drivers/oga_gfx.c	2020-11-15 16:18:56.826639877 -0500
+@@ -603,6 +603,12 @@ static bool oga_gfx_frame(void *data, co
+       oga_rect_t r;
+       oga_calc_bounds(&r, width, height, aspect_ratio);
+ 
++      settings_t *settings                 = config_get_ptr();
++      int rotation2 = HAL_TRANSFORM_ROT_270;
++
++      if (settings->bools.video_oga_vertical_enable)
++      rotation2 = 0;
++
+       oga_blit(vid->frame_surface, 0, 0, width, height,
+             page_surface, r.y, r.x, r.h, r.w, vid->rotation, vid->scale_mode, blend);
+    }
diff -rupN retroarch.orig/gfx/drivers_context/drm_go2_ctx.c retroarch.new/gfx/drivers_context/drm_go2_ctx.c
--- retroarch.orig/gfx/drivers_context/drm_go2_ctx.c	2021-01-03 07:37:05.039434346 -0500
+++ retroarch.new/gfx/drivers_context/drm_go2_ctx.c	2021-01-03 07:37:48.839856442 -0500
@@ -295,6 +295,7 @@ static bool gfx_ctx_go2_drm_suppress_scr
 
 static void gfx_ctx_go2_drm_swap_buffers(void *data)
 {
+   settings_t *settings                 = config_get_ptr();
    gfx_ctx_go2_drm_data_t 
       *drm   = (gfx_ctx_go2_drm_data_t*)data;
 
@@ -321,11 +322,21 @@ static void gfx_ctx_go2_drm_swap_buffers
    go2_context_swap_buffers(drm->context);
 
    go2_surface_t* surface = go2_context_surface_lock(drm->context);
+
+if (settings->bools.video_oga_vertical_enable) {
+   go2_presenter_post(drm->presenter,
+         surface,
+         src_x, src_y, src_w, src_h,
+         out_y, out_x, out_h, out_w,
+         GO2_ROTATION_DEGREES_0, 2);
+} else {
    go2_presenter_post(drm->presenter,
          surface,
          src_x, src_y, src_w, src_h,
          out_y, out_x, out_h, out_w,
          GO2_ROTATION_DEGREES_270, 2);
+}
+
    go2_context_surface_unlock(drm->context, surface);
 #endif
 }
