diff --git a/SDL/SDLMain.cpp b/SDL/SDLMain.cpp
index c172f439c1..cb86dec255 100644
--- a/SDL/SDLMain.cpp
+++ b/SDL/SDLMain.cpp
@@ -782,12 +782,6 @@ static void ProcessSDLEvent(SDL_Window *window, const SDL_Event &event, InputSta
 				g_Config.iWindowWidth = new_width;
 				g_Config.iWindowHeight = new_height;
 			}
-			// Hide/Show cursor correctly toggling fullscreen
-			if (lastUIState == UISTATE_INGAME && fullscreen && !g_Config.bShowTouchControls) {
-				SDL_ShowCursor(SDL_DISABLE);
-			} else if (lastUIState != UISTATE_INGAME || !fullscreen) {
-				SDL_ShowCursor(SDL_ENABLE);
-			}
 			break;
 		}
 
@@ -1500,7 +1494,7 @@ int main(int argc, char *argv[]) {
 				break;
 
 			UpdateTextFocus();
-			UpdateSDLCursor();
+			//UpdateSDLCursor();
 
 			inputTracker.MouseCaptureControl();
 
@@ -1527,7 +1521,7 @@ int main(int argc, char *argv[]) {
 			break;
 
 		UpdateTextFocus();
-		UpdateSDLCursor();
+		//UpdateSDLCursor();
 
 		inputTracker.MouseCaptureControl();
 
