diff -rupN mupen64plus-libretro.orig/GLideN64/src/ShaderUtils.h mupen64plus-libretro.new/GLideN64/src/ShaderUtils.h
--- mupen64plus-libretro.orig/GLideN64/src/ShaderUtils.h	2021-01-27 17:21:42.177358627 -0500
+++ mupen64plus-libretro.new/GLideN64/src/ShaderUtils.h	2021-01-27 17:22:02.081763415 -0500
@@ -1,6 +1,7 @@
 #ifndef SHADER_UTILS_H
 #define SHADER_UTILS_H
 
+#include <string>
 #include "OpenGL.h"
 #include "Combiner.h"
 
