diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 780b802..a0ec134 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -52,15 +52,13 @@ include(GNUInstallDirs)
 include(GetGitRevisionDescription)
 include(InstallRequiredSystemLibraries)
 include(FindPkgConfig)
-include(ExternalProject)
-include(BuildLibMPEG2)
 
 use_cxx11( )
 
 PKG_SEARCH_MODULE(SDL2 REQUIRED sdl2)
 PKG_SEARCH_MODULE(SDL2_TTF REQUIRED SDL2_ttf)
 PKG_SEARCH_MODULE(SDL2_MIXER REQUIRED SDL2_mixer)
-build_libmpeg2( )
+PKG_SEARCH_MODULE(MPEG2 REQUIRED libmpeg2)
 
 message(STATUS "Target: ${CMAKE_SYSTEM_NAME} ${CMAKE_TARGET_ARCHITECTURES}")
 
@@ -114,7 +112,6 @@ add_subdirectory(timer)
 add_subdirectory(video)
 add_subdirectory(vldp)
 
-add_dependencies( vldp libmpeg2 )
 add_dependencies( ldp-out vldp )
 add_dependencies( game vldp )
 add_dependencies( sound vldp )
@@ -162,3 +159,5 @@ set(CPACK_PACKAGE_VERSION_MINOR "${VER_MINOR}")
 set(CPACK_PACKAGE_VERSION_PATCH "${VER_PATCH}")
 
 include(CPack)
+install(TARGETS hypseus DESTINATION bin)
+install(DIRECTORY ../pics ../fonts ../sound DESTINATION /usr/share/daphne)
diff --git a/src/manymouse/CMakeLists.txt b/src/manymouse/CMakeLists.txt
index 7c3e66e..c5a8364 100644
--- a/src/manymouse/CMakeLists.txt
+++ b/src/manymouse/CMakeLists.txt
@@ -1,7 +1,8 @@
+
+
 set( LIB_SOURCES
     manymouse.c linux_evdev.c macosx_hidmanager.c
     macosx_hidutilities.c windows_wminput.c
-    x11_xinput2.c
 )
 
 set( LIB_HEADERS
diff --git a/src/manymouse/manymouse.c b/src/manymouse/manymouse.c
index 7f88582..8687e91 100644
--- a/src/manymouse/manymouse.c
+++ b/src/manymouse/manymouse.c
@@ -17,7 +17,6 @@ extern const ManyMouseDriver *ManyMouseDriver_windows;
 extern const ManyMouseDriver *ManyMouseDriver_evdev;
 extern const ManyMouseDriver *ManyMouseDriver_hidmanager;
 extern const ManyMouseDriver *ManyMouseDriver_hidutilities;
-extern const ManyMouseDriver *ManyMouseDriver_xinput2;
 
 /*
  * These have to be in the favored order...obviously it doesn't matter if the
@@ -31,7 +30,6 @@ extern const ManyMouseDriver *ManyMouseDriver_xinput2;
  */
 static const ManyMouseDriver **mice_drivers[] =
 {
-    &ManyMouseDriver_xinput2,
     &ManyMouseDriver_evdev,
     &ManyMouseDriver_windows,
     &ManyMouseDriver_hidmanager,
diff --git a/src/vldp/vldp.h b/src/vldp/vldp.h
index 438f1a1..a1498c5 100644
--- a/src/vldp/vldp.h
+++ b/src/vldp/vldp.h
@@ -35,6 +35,8 @@ extern "C" {
 // But CMake does the right thing here 
 #include <SDL.h> // only used for threading
 
+#include <mpeg2dec/mpeg2.h>
+
 struct yuv_buf {
     uint8_t *Y;     // Y channel
     uint8_t *U;     // U channel
diff --git a/src/vldp/vldp_internal.h b/src/vldp/vldp_internal.h
index 88450e9..7976e73 100644
--- a/src/vldp/vldp_internal.h
+++ b/src/vldp/vldp_internal.h
@@ -27,7 +27,6 @@
 
 #include "vldp.h" // for the VLDP_BOOL definition and SDL.h
 
-#include <mpeg2.h>
 
 // this is which version of the .dat file format we are using
 #define DAT_VERSION 3
