diff --git a/os_dep/linux/usb_intf.c b/os_dep/linux/usb_intf.c
index c10592b..ea58a77 100644
--- a/os_dep/linux/usb_intf.c
+++ b/os_dep/linux/usb_intf.c
@@ -1517,10 +1517,10 @@ int rockchip_wifi_init_module_rtkwifi(void)
 int rockchip_wifi_init_module(void)
 #endif
 {
-#ifdef CONFIG_WIFI_LOAD_DRIVER_WHEN_KERNEL_BOOTUP
-    int type = get_wifi_chip_type();
-    if (type < WIFI_AP6XXX_SERIES || type == WIFI_ESP8089) return 0;
-#endif
+// #ifdef CONFIG_WIFI_LOAD_DRIVER_WHEN_KERNEL_BOOTUP
+//     int type = get_wifi_chip_type();
+//     if (type < WIFI_AP6XXX_SERIES || type == WIFI_ESP8089) return 0;
+// #endif
     printk("\n");
     printk("=======================================================\n");
     printk("==== Launching Wi-Fi driver! (Powered by Rockchip) ====\n");
@@ -1544,10 +1544,10 @@ void rockchip_wifi_exit_module_rtkwifi(void)
 void rockchip_wifi_exit_module(void)
 #endif
 {
-#ifdef CONFIG_WIFI_LOAD_DRIVER_WHEN_KERNEL_BOOTUP
-    int type = get_wifi_chip_type();
-    if (type < WIFI_AP6XXX_SERIES || type == WIFI_ESP8089) return;
-#endif
+// #ifdef CONFIG_WIFI_LOAD_DRIVER_WHEN_KERNEL_BOOTUP
+//     int type = get_wifi_chip_type();
+//     if (type < WIFI_AP6XXX_SERIES || type == WIFI_ESP8089) return;
+// #endif
     printk("\n");
     printk("=======================================================\n");
     printk("==== Dislaunching Wi-Fi driver! (Powered by Rockchip) ====\n");
