diff --git a/menu/drivers/ozone.c b/menu/drivers/ozone.c
index cab4491028..97a614ff63 100644
--- a/menu/drivers/ozone.c
+++ b/menu/drivers/ozone.c
@@ -6837,7 +6837,7 @@ static void *ozone_init(void **userdata, bool video_is_threaded)
    ozone->last_width        = width;
    ozone->last_height       = height;
    ozone->last_scale_factor = gfx_display_get_dpi_scale(p_disp,
-         settings, width, height, false, false);
+         settings, width, height, false, false) * 2.000f;
 
    file_list_initialize(&ozone->selection_buf_old);
 
@@ -7902,7 +7902,7 @@ static void ozone_render(void *data,
    /* Check whether screen dimensions or menu scale
     * factor have changed */
    scale_factor = gfx_display_get_dpi_scale(p_disp, settings,
-         width, height, false, false);
+         width, height, false, false) * 2.000f;
 
    if ((scale_factor != ozone->last_scale_factor) ||
        (width != ozone->last_width) ||
