diff --git a/menu/drivers/ozone.c b/menu/drivers/ozone.c
index 2c26c12b6c..2f89a4f5c7 100644
--- a/menu/drivers/ozone.c
+++ b/menu/drivers/ozone.c
@@ -5203,20 +5203,20 @@ static void ozone_refresh_system_tabs_list(ozone_handle_t * ozone)
       ozone->tabs[++ozone->system_tab_end] = OZONE_SYSTEM_TAB_VIDEO;
 #endif
 
-   if (      settings->uints.menu_content_show_add_entry == MENU_ADD_CONTENT_ENTRY_DISPLAY_PLAYLISTS_TAB
-         && !settings->bools.kiosk_mode_enable)
-      ozone->tabs[++ozone->system_tab_end] = OZONE_SYSTEM_TAB_ADD;
-
-#if defined(HAVE_DYNAMIC)
-   if (settings->uints.menu_content_show_contentless_cores !=
-         MENU_CONTENTLESS_CORES_DISPLAY_NONE)
-      ozone->tabs[++ozone->system_tab_end] = OZONE_SYSTEM_TAB_CONTENTLESS_CORES;
-#endif
-
-#if defined(HAVE_LIBRETRODB)
-   if (settings->bools.menu_content_show_explore)
-      ozone->tabs[++ozone->system_tab_end] = OZONE_SYSTEM_TAB_EXPLORE;
-#endif
+   // if (      settings->uints.menu_content_show_add_entry == MENU_ADD_CONTENT_ENTRY_DISPLAY_PLAYLISTS_TAB
+         // && !settings->bools.kiosk_mode_enable)
+      // ozone->tabs[++ozone->system_tab_end] = OZONE_SYSTEM_TAB_ADD;
+
+// #if defined(HAVE_DYNAMIC)
+   // if (settings->uints.menu_content_show_contentless_cores !=
+         // MENU_CONTENTLESS_CORES_DISPLAY_NONE)
+      // ozone->tabs[++ozone->system_tab_end] = OZONE_SYSTEM_TAB_CONTENTLESS_CORES;
+// #endif
+
+// #if defined(HAVE_LIBRETRODB)
+   // if (settings->bools.menu_content_show_explore)
+      // ozone->tabs[++ozone->system_tab_end] = OZONE_SYSTEM_TAB_EXPLORE;
+// #endif
 }
 
 static int ozone_get_entries_padding_old_list(ozone_handle_t* ozone)
diff --git a/menu/menu_setting.c b/menu/menu_setting.c
index 6371398a27..916511a7b2 100644
--- a/menu/menu_setting.c
+++ b/menu/menu_setting.c
@@ -9199,7 +9199,7 @@ static void update_streaming_url_write_handler(rarch_setting_t *setting)
    recording_driver_update_streaming_url();
 }
 
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 static void systemd_service_toggle(const char *path, char *unit, bool enable)
 {
    pid_t pid    = fork();
@@ -10301,7 +10301,7 @@ static bool setting_append_list(
 #endif
 #if !defined(IOS)
          /* Apple rejects iOS apps that let you forcibly quit them. */
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
          CONFIG_ACTION(
                list, list_info,
                MENU_ENUM_LABEL_QUIT_RETROARCH,
@@ -10331,7 +10331,7 @@ static bool setting_append_list(
               parent_group);
 #endif
 
-#if defined(HAVE_LAKKA)
+#if !defined(HAVE_LAKKA)
          CONFIG_ACTION(
                list, list_info,
                MENU_ENUM_LABEL_REBOOT,
@@ -10699,7 +10699,7 @@ static bool setting_append_list(
                &group_info,
                &subgroup_info,
                parent_group);
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
          SETTINGS_DATA_LIST_CURRENT_ADD_FLAGS(list, list_info, SD_FLAG_ADVANCED);
 #endif
 
@@ -10717,7 +10717,7 @@ static bool setting_append_list(
          }
 #endif
 
-#if defined(HAVE_LAKKA) || defined(HAVE_WIFI)
+#if !defined(HAVE_LAKKA) || defined(HAVE_WIFI)
          if (string_is_not_equal(settings->arrays.wifi_driver, "null"))
          {
             CONFIG_ACTION(
@@ -10739,7 +10739,7 @@ static bool setting_append_list(
                parent_group);
          SETTINGS_DATA_LIST_CURRENT_ADD_FLAGS(list, list_info, SD_FLAG_LAKKA_ADVANCED);
 
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
          CONFIG_ACTION(
                list, list_info,
                MENU_ENUM_LABEL_LAKKA_SERVICES,
@@ -19363,7 +19363,7 @@ static bool setting_append_list(
                   general_read_handler,
                   SD_FLAG_LAKKA_ADVANCED);
 
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
             CONFIG_BOOL(
                   list, list_info,
                   &settings->bools.menu_show_quit_retroarch,
@@ -20587,7 +20587,7 @@ static bool setting_append_list(
                SD_FLAG_CMD_APPLY_AUTO);
 #endif
 
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
          CONFIG_ACTION(
                list, list_info,
                MENU_ENUM_LABEL_CPU_PERFPOWER,
@@ -23466,7 +23466,7 @@ static bool setting_append_list(
          break;
       case SETTINGS_LIST_LAKKA_SERVICES:
          {
-#if defined(HAVE_LAKKA)
+#if !defined(HAVE_LAKKA)
             START_GROUP(list, list_info, &group_info,
                   msg_hash_to_str(MENU_ENUM_LABEL_VALUE_LAKKA_SERVICES),
                   parent_group);
