diff --git a/input/drivers/udev_input.c b/input/drivers/udev_input.c
index 489e00fa9a..4b4d1a05aa 100644
--- a/input/drivers/udev_input.c
+++ b/input/drivers/udev_input.c
@@ -81,7 +81,7 @@
 #endif
 
 /* Force UDEV_XKB_HANDLING for Lakka */
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 #ifndef UDEV_XKB_HANDLING
 #define UDEV_XKB_HANDLING
 #endif
@@ -1418,7 +1418,7 @@ static void *udev_input_init(const char *joypad_driver)
       goto error;
 
    video_context_driver_get_ident(&ctx_ident);
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
    /* Force xkb_handling on Lakka */
    udev->xkb_handling = true;
 #else
