diff --git a/deps/libgo2/src/display.c b/deps/libgo2/src/display.c
index b91cf0624b..9488a6c279 100644
--- a/deps/libgo2/src/display.c
+++ b/deps/libgo2/src/display.c
@@ -550,7 +550,7 @@ static uint32_t go2_rkformat_get(uint32_t drm_fourcc)
 
         case DRM_FORMAT_ARGB8888:
         case DRM_FORMAT_XRGB8888:
-            return RK_FORMAT_BGRA_8888;
+            return RK_FORMAT_RGBA_8888;
 
         case DRM_FORMAT_RGB565:
             return RK_FORMAT_RGB_565;
diff --git a/gfx/drivers_context/drm_go2_ctx.c b/gfx/drivers_context/drm_go2_ctx.c
index 856aa0f9a2..5d607408c5 100644
--- a/gfx/drivers_context/drm_go2_ctx.c
+++ b/gfx/drivers_context/drm_go2_ctx.c
@@ -131,7 +131,7 @@ static void *gfx_ctx_go2_drm_init(void *video_driver)
    }
 
    drm->presenter     = go2_presenter_create(drm->display,
-         DRM_FORMAT_RGB565, 0xff000000, true);
+         DRM_FORMAT_XRGB8888, 0xff000000, true);
 
    return drm;
 }
