diff --git a/menu/menu_displaylist.c b/menu/menu_displaylist.c
index 2a7db944af..037c820f5c 100644
--- a/menu/menu_displaylist.c
+++ b/menu/menu_displaylist.c
@@ -14669,7 +14669,7 @@ bool menu_displaylist_ctl(enum menu_displaylist_ctl_state type,
                flags.flags     = 0;
                video_context_driver_get_flags(&flags);
 #endif
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
                if (menu_entries_append(info->list,
                         msg_hash_to_str(MENU_ENUM_LABEL_VALUE_UPDATE_LAKKA),
                         msg_hash_to_str(MENU_ENUM_LABEL_UPDATE_LAKKA),
diff --git a/menu/menu_setting.c b/menu/menu_setting.c
index 589cbe7f49..25d5284918 100644
--- a/menu/menu_setting.c
+++ b/menu/menu_setting.c
@@ -10563,7 +10563,7 @@ static bool setting_append_list(
 #endif
 #if !defined(IOS)
          /* Apple rejects iOS apps that let you forcibly quit them. */
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
          CONFIG_ACTION(
                list, list_info,
                MENU_ENUM_LABEL_QUIT_RETROARCH,
