diff --git a/menu/menu_displaylist.c b/menu/menu_displaylist.c
index e931c3238d..4a3bfe023e 100644
--- a/menu/menu_displaylist.c
+++ b/menu/menu_displaylist.c
@@ -14107,52 +14107,52 @@ bool menu_displaylist_ctl(enum menu_displaylist_ctl_state type,
 #endif /* HAVE_UPDATE_CORES */
 
 #if defined(HAVE_COMPRESSION) && !defined(HAVE_MIST)
-               if (menu_entries_append(info->list,
-                        msg_hash_to_str(MENU_ENUM_LABEL_VALUE_DOWNLOAD_CORE_SYSTEM_FILES),
-                        msg_hash_to_str(MENU_ENUM_LABEL_DOWNLOAD_CORE_SYSTEM_FILES),
-                        MENU_ENUM_LABEL_DOWNLOAD_CORE_SYSTEM_FILES,
-                        MENU_SETTING_ACTION, 0, 0, NULL))
-                  count++;
+               //if (menu_entries_append(info->list,
+               //         msg_hash_to_str(MENU_ENUM_LABEL_VALUE_DOWNLOAD_CORE_SYSTEM_FILES),
+               //         msg_hash_to_str(MENU_ENUM_LABEL_DOWNLOAD_CORE_SYSTEM_FILES),
+               //         MENU_ENUM_LABEL_DOWNLOAD_CORE_SYSTEM_FILES,
+               //         MENU_SETTING_ACTION, 0, 0, NULL))
+               //   count++;
 #endif
 
-               if (menu_entries_append(info->list,
-                        msg_hash_to_str(MENU_ENUM_LABEL_VALUE_DOWNLOAD_CORE_CONTENT),
-                        msg_hash_to_str(MENU_ENUM_LABEL_DOWNLOAD_CORE_CONTENT_DIRS),
-                        MENU_ENUM_LABEL_DOWNLOAD_CORE_CONTENT_DIRS,
-                        MENU_SETTING_ACTION, 0, 0, NULL))
-                  count++;
+               //if (menu_entries_append(info->list,
+               //         msg_hash_to_str(MENU_ENUM_LABEL_VALUE_DOWNLOAD_CORE_CONTENT),
+               //         msg_hash_to_str(MENU_ENUM_LABEL_DOWNLOAD_CORE_CONTENT_DIRS),
+               //         MENU_ENUM_LABEL_DOWNLOAD_CORE_CONTENT_DIRS,
+               //         MENU_SETTING_ACTION, 0, 0, NULL))
+               //   count++;
 
 #ifdef HAVE_ONLINE_UPDATER
-               if (MENU_DISPLAYLIST_PARSE_SETTINGS_ENUM(info->list,
-                     MENU_ENUM_LABEL_UPDATER_SETTINGS,
-                     PARSE_ACTION, false) == 0)
-                  count++;
+               //if (MENU_DISPLAYLIST_PARSE_SETTINGS_ENUM(info->list,
+               //      MENU_ENUM_LABEL_UPDATER_SETTINGS,
+               //      PARSE_ACTION, false) == 0)
+               //   count++;
 #endif
 
-               if (menu_entries_append(info->list,
-                        msg_hash_to_str(MENU_ENUM_LABEL_VALUE_PL_THUMBNAILS_UPDATER_LIST),
-                        msg_hash_to_str(MENU_ENUM_LABEL_PL_THUMBNAILS_UPDATER_LIST),
-                        MENU_ENUM_LABEL_PL_THUMBNAILS_UPDATER_LIST,
-                        MENU_SETTING_ACTION, 0, 0, NULL))
-                  count++;
+               //if (menu_entries_append(info->list,
+               //         msg_hash_to_str(MENU_ENUM_LABEL_VALUE_PL_THUMBNAILS_UPDATER_LIST),
+               //         msg_hash_to_str(MENU_ENUM_LABEL_PL_THUMBNAILS_UPDATER_LIST),
+               //         MENU_ENUM_LABEL_PL_THUMBNAILS_UPDATER_LIST,
+               //         MENU_SETTING_ACTION, 0, 0, NULL))
+               //   count++;
 
 #if 0
 /* Thumbnailpack removal */
-               if (settings->bools.menu_show_legacy_thumbnail_updater)
-               {
-                  if (menu_entries_append(info->list,
-                           msg_hash_to_str(MENU_ENUM_LABEL_VALUE_THUMBNAILS_UPDATER_LIST),
-                           msg_hash_to_str(MENU_ENUM_LABEL_THUMBNAILS_UPDATER_LIST),
-                           MENU_ENUM_LABEL_THUMBNAILS_UPDATER_LIST,
-                           MENU_SETTING_ACTION, 0, 0, NULL))
-                     count++;
-               }
+               //if (settings->bools.menu_show_legacy_thumbnail_updater)
+               //{
+               //   if (menu_entries_append(info->list,
+               //            msg_hash_to_str(MENU_ENUM_LABEL_VALUE_THUMBNAILS_UPDATER_LIST),
+               //            msg_hash_to_str(MENU_ENUM_LABEL_THUMBNAILS_UPDATER_LIST),
+               //            MENU_ENUM_LABEL_THUMBNAILS_UPDATER_LIST,
+               //            MENU_SETTING_ACTION, 0, 0, NULL))
+               //      count++;
+               //}
 #endif
 #ifdef HAVE_NETWORKING
-               if (MENU_DISPLAYLIST_PARSE_SETTINGS_ENUM(info->list,
-                        MENU_ENUM_LABEL_NETWORK_ON_DEMAND_THUMBNAILS,
-                        PARSE_ONLY_BOOL, false) != -1)
-                  count++;
+               //if (MENU_DISPLAYLIST_PARSE_SETTINGS_ENUM(info->list,
+               //         MENU_ENUM_LABEL_NETWORK_ON_DEMAND_THUMBNAILS,
+               //         PARSE_ONLY_BOOL, false) != -1)
+               //   count++;
 #endif
 
 #ifdef HAVE_COMPRESSION
@@ -14169,20 +14169,20 @@ bool menu_displaylist_ctl(enum menu_displaylist_ctl_state type,
 #endif
 
 #ifdef HAVE_UPDATE_ASSETS
-               if (menu_entries_append(info->list,
-                        msg_hash_to_str(MENU_ENUM_LABEL_VALUE_UPDATE_ASSETS),
-                        msg_hash_to_str(MENU_ENUM_LABEL_UPDATE_ASSETS),
-                        MENU_ENUM_LABEL_UPDATE_ASSETS,
-                        MENU_SETTING_ACTION, 0, 0, NULL))
-                  count++;
+               //if (menu_entries_append(info->list,
+               //         msg_hash_to_str(MENU_ENUM_LABEL_VALUE_UPDATE_ASSETS),
+               //         msg_hash_to_str(MENU_ENUM_LABEL_UPDATE_ASSETS),
+               //         MENU_ENUM_LABEL_UPDATE_ASSETS,
+               //         MENU_SETTING_ACTION, 0, 0, NULL))
+               //   count++;
 #endif
 #if !defined(_3DS)
-               if (menu_entries_append(info->list,
-                        msg_hash_to_str(MENU_ENUM_LABEL_VALUE_UPDATE_AUTOCONFIG_PROFILES),
-                        msg_hash_to_str(MENU_ENUM_LABEL_UPDATE_AUTOCONFIG_PROFILES),
-                        MENU_ENUM_LABEL_UPDATE_AUTOCONFIG_PROFILES,
-                        MENU_SETTING_ACTION, 0, 0, NULL))
-                  count++;
+               //if (menu_entries_append(info->list,
+               //         msg_hash_to_str(MENU_ENUM_LABEL_VALUE_UPDATE_AUTOCONFIG_PROFILES),
+               //         msg_hash_to_str(MENU_ENUM_LABEL_UPDATE_AUTOCONFIG_PROFILES),
+               //         MENU_ENUM_LABEL_UPDATE_AUTOCONFIG_PROFILES,
+               //         MENU_SETTING_ACTION, 0, 0, NULL))
+               //   count++;
 
                if (menu_entries_append(info->list,
                         msg_hash_to_str(MENU_ENUM_LABEL_VALUE_UPDATE_CHEATS),
@@ -14193,21 +14193,21 @@ bool menu_displaylist_ctl(enum menu_displaylist_ctl_state type,
 #endif
 #ifdef HAVE_LIBRETRODB
 #if !defined(VITA)
-               if (menu_entries_append(info->list,
-                        msg_hash_to_str(MENU_ENUM_LABEL_VALUE_UPDATE_DATABASES),
-                        msg_hash_to_str(MENU_ENUM_LABEL_UPDATE_DATABASES),
-                        MENU_ENUM_LABEL_UPDATE_DATABASES,
-                        MENU_SETTING_ACTION, 0, 0, NULL))
-                  count++;
+               //if (menu_entries_append(info->list,
+               //         msg_hash_to_str(MENU_ENUM_LABEL_VALUE_UPDATE_DATABASES),
+               //         msg_hash_to_str(MENU_ENUM_LABEL_UPDATE_DATABASES),
+               //         MENU_ENUM_LABEL_UPDATE_DATABASES,
+               //         MENU_SETTING_ACTION, 0, 0, NULL))
+               //   count++;
 #endif
 #endif /* HAVE_LIBRETRODB */
 #if !defined(_3DS)
-               if (menu_entries_append(info->list,
-                        msg_hash_to_str(MENU_ENUM_LABEL_VALUE_UPDATE_OVERLAYS),
-                        msg_hash_to_str(MENU_ENUM_LABEL_UPDATE_OVERLAYS),
-                        MENU_ENUM_LABEL_UPDATE_OVERLAYS,
-                        MENU_SETTING_ACTION, 0, 0, NULL))
-                  count++;
+               //if (menu_entries_append(info->list,
+               //         msg_hash_to_str(MENU_ENUM_LABEL_VALUE_UPDATE_OVERLAYS),
+               //         msg_hash_to_str(MENU_ENUM_LABEL_UPDATE_OVERLAYS),
+               //         MENU_ENUM_LABEL_UPDATE_OVERLAYS,
+               //         MENU_SETTING_ACTION, 0, 0, NULL))
+               //   count++;
 
 #if defined(HAVE_CG) || defined(HAVE_GLSL) || defined(HAVE_SLANG) || defined(HAVE_HLSL)
                if (video_shader_is_supported(RARCH_SHADER_CG))
@@ -14220,15 +14220,15 @@ bool menu_displaylist_ctl(enum menu_displaylist_ctl_state type,
                      count++;
                }
 
-               if (video_shader_is_supported(RARCH_SHADER_GLSL))
-               {
-                  if (menu_entries_append(info->list,
-                           msg_hash_to_str(MENU_ENUM_LABEL_VALUE_UPDATE_GLSL_SHADERS),
-                           msg_hash_to_str(MENU_ENUM_LABEL_UPDATE_GLSL_SHADERS),
-                           MENU_ENUM_LABEL_UPDATE_GLSL_SHADERS,
-                           MENU_SETTING_ACTION, 0, 0, NULL))
-                     count++;
-               }
+               //if (video_shader_is_supported(RARCH_SHADER_GLSL))
+               //{
+               //   if (menu_entries_append(info->list,
+               //            msg_hash_to_str(MENU_ENUM_LABEL_VALUE_UPDATE_GLSL_SHADERS),
+               //            msg_hash_to_str(MENU_ENUM_LABEL_UPDATE_GLSL_SHADERS),
+               //            MENU_ENUM_LABEL_UPDATE_GLSL_SHADERS,
+               //            MENU_SETTING_ACTION, 0, 0, NULL))
+               //      count++;
+               //}
 
                if (video_shader_is_supported(RARCH_SHADER_SLANG))
                {
