diff -rupN jslisten.orig/src/jslisten.c jslisten.new/src/jslisten.c
--- jslisten.orig/src/jslisten.c	2021-01-31 20:22:41.960845892 -0500
+++ jslisten.new/src/jslisten.c	2021-01-31 20:23:51.734140471 -0500
@@ -53,9 +53,9 @@
 
 #define NAME_LENGTH 128
 #define MYPROGNAME "jslisten"
-#define myConfFile "/.kodi/addons/script.emuelec.Amlogic.launcher/config/jslisten.cfg"
-#define myConfFileNG "/.kodi/addons/script.emuelec.Amlogic-ng.launcher/config/jslisten.cfg"
-#define myGlConfFile "/emuelec/configs/jslisten.cfg"
+#define myConfFile "/storage/.config/distribution/configs/jslisten.cfg"
+#define myConfFileNG "/storage/.config/distribution/configs/jslisten.cfg"
+#define myGlConfFile "/storage/.config/distribution/configs/jslisten.cfg"
 //#define MY_LOG_LEVEL LOG_NOTICE //LOG_DEBUG //LOG_NOTICE
 
 #define INI_BUFFERSIZE      512
