diff -rupN u-boot.orig/board/rockchip/odroidgo2/odroidgo2.c u-boot.new/board/rockchip/odroidgo2/odroidgo2.c
--- u-boot.orig/board/rockchip/odroidgo2/odroidgo2.c	2021-04-16 18:25:14.604370207 -0400
+++ u-boot.new/board/rockchip/odroidgo2/odroidgo2.c	2021-04-16 19:47:59.072051638 -0400
@@ -184,9 +184,9 @@ int rk_board_late_init(void)
 	/* show boot logo and version : drivers/video/drm/rockchip_display_cmds.c */
 	lcd_show_logo();
 	lcd_setfg_color("white");
-	lcd_printf(0, 18, 1, "%s", U_BOOT_VERSION);
-	lcd_printf(0, 19, 1, "%s %s", U_BOOT_DATE, U_BOOT_TIME);
-
+	lcd_printf(0, 16, 1, "%s %s", U_BOOT_DATE, U_BOOT_TIME);
+	lcd_printf(0, 18, 1, "This software can not be bundled or sold!");
+	lcd_printf(0, 19, 1, "If you paid for it, demand a refund!");
 	if (!board_check_autotest()) {
 		board_run_autotest();
 		return 0;
