diff -rupN RG351P-linux-718abd7245ab07d144e03826d7b0f161ab66bed2.orig/arch/arm64/boot/dts/rockchip/px30.dtsi RG351P-linux-718abd7245ab07d144e03826d7b0f161ab66bed2/arch/arm64/boot/dts/rockchip/px30.dtsi
--- RG351P-linux-718abd7245ab07d144e03826d7b0f161ab66bed2.orig/arch/arm64/boot/dts/rockchip/px30.dtsi	2020-10-09 23:27:58.000000000 -0400
+++ RG351P-linux-718abd7245ab07d144e03826d7b0f161ab66bed2/arch/arm64/boot/dts/rockchip/px30.dtsi	2020-12-16 16:26:17.334749792 -0500
@@ -111,10 +111,10 @@
 		rockchip,temp-hysteresis = <5000>;
 		rockchip,low-temp = <0>;
 		rockchip,low-temp-min-volt = <1000000>;
-		rockchip,low-temp-adjust-volt = <
-			/* MHz    MHz    uV */
-			0      1512   50000
-		>;
+//		rockchip,low-temp-adjust-volt = <
+//			/* MHz    MHz    uV */
+//			0      1512   50000
+//		>;
 
 		clocks = <&cru PLL_APLL>;
 		rockchip,avs-scale = <4>;
@@ -207,24 +207,33 @@
 			opp-microvolt-L3 = <1250000 1250000 1350000>;
 			clock-latency-ns = <40000>;
 		};
-		opp-1416000000 {
-			opp-hz = /bits/ 64 <1416000000>;
-			opp-microvolt = <1350000 1350000 1350000>;
-			opp-microvolt-L0 = <1350000 1350000 1350000>;
-			opp-microvolt-L1 = <1350000 1350000 1350000>;
-			opp-microvolt-L2 = <1300000 1300000 1350000>;
-			opp-microvolt-L3 = <1250000 1250000 1350000>;
-			clock-latency-ns = <40000>;
-		};
-		opp-1512000000 {
-			opp-hz = /bits/ 64 <1512000000>;
-			opp-microvolt = <1350000 1350000 1350000>;
-			opp-microvolt-L0 = <1350000 1350000 1350000>;
-			opp-microvolt-L1 = <1350000 1350000 1350000>;
-			opp-microvolt-L2 = <1300000 1300000 1350000>;
-			opp-microvolt-L3 = <1250000 1250000 1350000>;
-			clock-latency-ns = <40000>;
-		};
+                opp-1368000000 {
+                        opp-hz = /bits/ 64 <1368000000>;
+                        opp-microvolt = <1350000 1350000 1350000>;
+                        opp-microvolt-L0 = <1350000 1350000 1350000>;
+                        opp-microvolt-L1 = <1350000 1350000 1350000>;
+                        opp-microvolt-L2 = <1300000 1300000 1350000>;
+                        opp-microvolt-L3 = <1250000 1250000 1350000>;
+                        clock-latency-ns = <40000>;
+                };
+		//opp-1416000000 {
+		//	opp-hz = /bits/ 64 <1416000000>;
+                //        opp-microvolt = <1350000 1350000 1350000>;
+                //        opp-microvolt-L0 = <1350000 1350000 1350000>;
+                //        opp-microvolt-L1 = <1350000 1350000 1350000>;
+                //        opp-microvolt-L2 = <1300000 1300000 1350000>;
+                //        opp-microvolt-L3 = <1250000 1250000 1350000>;
+                //        clock-latency-ns = <40000>;
+		//};
+                opp-1512000000 {
+                        opp-hz = /bits/ 64 <1512000000>;
+                        opp-microvolt = <1350000 1350000 1350000>;
+                        opp-microvolt-L0 = <1350000 1350000 1350000>;
+                        opp-microvolt-L1 = <1350000 1350000 1350000>;
+                        opp-microvolt-L2 = <1300000 1300000 1350000>;
+                        opp-microvolt-L3 = <1250000 1250000 1350000>;
+                        clock-latency-ns = <40000>;
+                };
 	};
 
 	arm-pmu {
@@ -1354,10 +1363,10 @@
 		rockchip,temp-hysteresis = <5000>;
 		rockchip,low-temp = <0>;
 		rockchip,low-temp-min-volt = <1000000>;
-		rockchip,low-temp-adjust-volt = <
-			/* MHz    MHz    uV */
-			0      480     50000
-		>;
+//		rockchip,low-temp-adjust-volt = <
+//			/* MHz    MHz    uV */
+//			0      480     50000
+//		>;
 
 		rockchip,max-volt = <1175000>;
 		rockchip,evb-irdrop = <25000>;
@@ -1827,16 +1836,16 @@
 		ddr_timing = <&ddr_timing>;
 		upthreshold = <40>;
 		downdifferential = <20>;
-		system-status-freq = <
-			/*system status         freq(KHz)*/
-			SYS_STATUS_NORMAL       528000
-			SYS_STATUS_REBOOT       450000
-			SYS_STATUS_SUSPEND      194000
-			SYS_STATUS_VIDEO_1080P  450000
-			SYS_STATUS_BOOST        528000
-			SYS_STATUS_ISP          666000
-			SYS_STATUS_PERFORMANCE  666000
-		>;
+                system-status-freq = <
+                        /*system status         freq(KHz)*/
+                        SYS_STATUS_NORMAL       528000
+                        SYS_STATUS_REBOOT       450000
+                        SYS_STATUS_SUSPEND      194000
+                        SYS_STATUS_VIDEO_1080P  450000
+                        SYS_STATUS_BOOST        850000
+                        SYS_STATUS_ISP          850000
+                        SYS_STATUS_PERFORMANCE  850000
+                >;
 		auto-min-freq = <328000>;
 		auto-freq-en = <1>;
 		#cooling-cells = <2>;
@@ -1914,6 +1923,10 @@
 			opp-microvolt-L3 = <1000000>;
 			status = "disabled";
 		};
+                opp-850000000 {
+                        opp-hz = /bits/ 64 <850000000>;
+                        opp-microvolt = <1150000>;
+                };
 	};
 
 	rockchip_system_monitor: rockchip-system-monitor {
diff -rupN RG351P-linux-718abd7245ab07d144e03826d7b0f161ab66bed2.orig/arch/arm64/boot/dts/rockchip/rk3326-rg351p-linux.dts RG351P-linux-718abd7245ab07d144e03826d7b0f161ab66bed2/arch/arm64/boot/dts/rockchip/rk3326-rg351p-linux.dts
--- RG351P-linux-718abd7245ab07d144e03826d7b0f161ab66bed2.orig/arch/arm64/boot/dts/rockchip/rk3326-rg351p-linux.dts	2020-10-09 23:27:58.000000000 -0400
+++ RG351P-linux-718abd7245ab07d144e03826d7b0f161ab66bed2/arch/arm64/boot/dts/rockchip/rk3326-rg351p-linux.dts	2020-12-16 16:33:09.919358699 -0500
@@ -146,16 +146,17 @@
 &cpu0_opp_table {
 	rockchip,avs = <1>;
 
-	/* bin scaling clock 1.296MHz */
 	rockchip,bin-scaling-sel = <
-		0               13
-		1               13
+	  0	10
+	  1	13
+	  2	16
+	  3	20
 	>;
 
 	/delete-node/ opp-408000000;
 	/delete-node/ opp-600000000;
 	/delete-node/ opp-816000000;
-	/delete-node/ opp-1416000000;
+	/delete-node/ opp-1248000000;
 	/delete-node/ opp-1512000000;
 };
 
@@ -423,7 +424,7 @@
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <950000>;
-				regulator-max-microvolt = <1350000>;
+				regulator-max-microvolt = <1400000>;
 				regulator-ramp-delay = <6001>;
 				regulator-initial-mode = <0x2>;
 				regulator-name = "vdd_arm";
