--- a/es-app/src/guis/GuiMenu.cpp.orig	2020-11-10 11:17:12.750020361 -0500
+++ b/es-app/src/guis/GuiMenu.cpp	2020-11-10 11:18:37.458344913 -0500
@@ -109,7 +109,7 @@ GuiMenu::GuiMenu(Window *window, bool an
 
 #ifdef _ENABLEEMUELEC
 	if (isFullUI)
-		addEntry(_("EMUELEC SETTINGS").c_str(), true, [this] { openEmuELECSettings(); }, "iconEmuelec"); /* < emuelec */
+		addEntry(_("351ELEC SETTINGS").c_str(), true, [this] { openEmuELECSettings(); }, "iconEmuelec"); /* < emuelec */
 #endif
 
 	if (isFullUI &&
@@ -683,7 +683,7 @@ void GuiMenu::addVersionInfo()
 		else
 #endif
 #ifdef _ENABLEEMUELEC	
-		mVersion.setText("EMUELEC ES V" + ApiSystem::getInstance()->getVersion() + buildDate + " IP:" + getShOutput(R"(/emuelec/scripts/emuelec-utils getip)"));
+		mVersion.setText("351ELEC ES V" + ApiSystem::getInstance()->getVersion() + buildDate + " IP:" + getShOutput(R"(/emuelec/scripts/emuelec-utils getip)"));
 #else
 		mVersion.setText("BATOCERA.LINUX ES V" + ApiSystem::getInstance()->getVersion() + buildDate);
 #endif
