diff --git a/Makefile.libretro b/Makefile.libretro
index 4721d35..650ae13 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -78,8 +78,6 @@ else ifeq ($(platform), rpi3)
 	USE_PICASSO96 = 1
 	CFLAGS += $(PLATFORM_DEFINES)
 	CXXFLAGS += $(PLATFORM_DEFINES)
-	CC = gcc
-	CXX = g++ 
 	LDFLAGS += -lz -lpthread -lFLAC -lmpg123 -ldl
 else ifeq ($(platform), rpi2)
 	TARGET := $(TARGET_NAME)_libretro.so
