diff -rupN dosbox-pure-a23d793ffbb89a4471fcf192fcbfffa5b04c745e.orig/Makefile dosbox-pure-a23d793ffbb89a4471fcf192fcbfffa5b04c745e.new/Makefile
--- dosbox-pure-a23d793ffbb89a4471fcf192fcbfffa5b04c745e.orig/Makefile	2020-12-28 20:14:11.147561757 -0500
+++ dosbox-pure-a23d793ffbb89a4471fcf192fcbfffa5b04c745e.new/Makefile	2020-12-28 20:42:27.635227311 -0500
@@ -19,7 +19,7 @@
 ISWIN      := $(findstring :,$(firstword $(subst \, ,$(subst /, ,$(abspath .)))))
 ISMAC      := $(wildcard /Applications)
 PIPETONULL := $(if $(ISWIN),>nul 2>nul,>/dev/null 2>/dev/null)
-PROCCPU    := $(shell $(if $(ISWIN),GenuineIntel Intel sse sse2,cat /proc/cpuinfo))
+#PROCCPU    := $(shell $(if $(ISWIN),GenuineIntel Intel sse sse2,cat /proc/cpuinfo))
 
 SOURCES := \
 	*.cpp       \
@@ -27,60 +27,20 @@ SOURCES := \
 	src/*/*.cpp \
 	src/*/*/*.cpp
 
-ifneq ($(ISWIN),)
-  OUTNAME := dosbox_pure_libretro.dll
-  CXX     := g++
-  LDFLAGS :=  -Wl,--gc-sections
-else ifneq ($(ISMAC),)
-  OUTNAME := dosbox_pure_libretro.dylib
-  CXX     := clang++
-  LDFLAGS :=  -Wl,-dead_strip
-else
-  OUTNAME := dosbox_pure_libretro.so
-  CXX     := g++
-  LDFLAGS :=  -Wl,--gc-sections
-endif
-
-ifneq ($(and $(filter ARMv7,$(PROCCPU)),$(filter neon,$(PROCCPU))),)
-  CPUFLAGS := -mcpu=cortex-a72 -mfpu=neon-fp-armv8 -mfloat-abi=hard -ffast-math
-  ifneq ($(findstring version 10,$(shell g++ -v 2>&1)),)
-    # Switch to gcc 9 to avoid buggy assembly genetation of gcc 10
-    # On armv7l, gcc 10.2 with -O2 on the file core_dynrec.cpp generates assembly that wrongfully passes NULL to the runcode function
-    # resulting in a segfault crash. It can be observed by writing block->cache.start to stdout twice where it is NULL at first
-    # and then the actual value thereafter. This affects upstream SVN DOSBox as well as this core.
-    CXX := g++-9
-  endif
-else
-  CPUFLAGS :=
-endif
-
-ifeq ($(BUILD),DEBUG)
-  BUILDDIR := debug
-  CFLAGS   := -DDEBUG -D_DEBUG -g -O0
-else ifeq ($(BUILD),PROFILE)
-  BUILDDIR := profile
-  CFLAGS   := -DNDEBUG -O2
-else ifeq ($(BUILD),RELEASEDBG)
-  BUILDDIR := releasedbg
-  CFLAGS   := -DNDEBUG -ggdb -O2
-  LDFLAGS  += -ggdb -O2
-else ifeq ($(BUILD),ASAN)
-  BUILDDIR := asan
-  CFLAGS   := -DDEBUG -D_DEBUG -g -O0 -fsanitize=address -fno-omit-frame-pointer
-  LDFLAGS  += -fsanitize=address -g -O0
-else
-  BUILD    := RELEASE
-  BUILDDIR := release
-  CFLAGS   := -DNDEBUG -O2 -fno-ident
-  LDFLAGS  += -O2 -fno-ident
-endif
+CPUFLAGS := -mtune=cortex-a35 -mcpu=cortex-a35 -march=armv8-a+crc+fp+simd
 
-CFLAGS  += $(CPUFLAGS) -std=c++11 -fpic -fomit-frame-pointer -fno-exceptions -fno-non-call-exceptions -Wno-address-of-packed-member -Wno-format -Wno-switch
+OUTNAME := dosbox_pure_libretro.so
+
+BUILD    := RELEASE
+BUILDDIR := release
+CFLAGS   := -DNDEBUG -O2 -fno-ident
+LDFLAGS  += -O2 -fno-ident
+
+CFLAGS  += $(CPUFLAGS) -fpic -fomit-frame-pointer -fno-exceptions -fno-non-call-exceptions -Wno-psabi -Wno-format
 CFLAGS  += -fvisibility=hidden -ffunction-sections -fdata-sections
 CFLAGS  += -pthread -D__LIBRETRO__ -Iinclude
 
-LDFLAGS += $(CPUFLAGS) -lpthread -shared
-#LDFLAGS += -static-libstdc++ -static-libgcc #adds 1MB to output
+LDFLAGS += $(CPUFLAGS) -lpthread -Wl,--gc-sections -shared
 
 .PHONY: all clean
 all: $(OUTNAME)
