diff --git a/core_options.h b/core_options.h
index 8b22b63..0df3e0d 100644
--- a/core_options.h
+++ b/core_options.h
@@ -451,21 +451,6 @@ static retro_core_option_v2_definition option_defs[] =
 		},
 		"svga_s3"
 	},
-	{
-		"dosbox_pure_svgamem",
-		"SVGA Memory (restart required)", NULL,
-		"The amount of memory available to the emulated SVGA card.", NULL,
-		"Video",
-		{
-			{ "0",  "512KB" },
-			{ "1", "1MB" },
-			{ "2", "2MB (default)" },
-			{ "3", "3MB" },
-			{ "4", "4MB" },
-			{ "8", "8MB (not always recognized)" },
-		},
-		"2"
-	},
 	{
 		"dosbox_pure_voodoo",
 		"3dfx Voodoo Emulation", NULL,
diff --git a/dosbox_pure_libretro.cpp b/dosbox_pure_libretro.cpp
index ab60c56..d4750cf 100644
--- a/dosbox_pure_libretro.cpp
+++ b/dosbox_pure_libretro.cpp
@@ -2847,7 +2847,6 @@ static bool check_variables(bool is_startup = false)
 		Variables::DosBoxSet("serial", "serial1", ((retro_get_variable("dosbox_pure_modem", "null")[0] == 'n') ? "libretro null" : "libretro"));
 
 	retro_set_visibility("dosbox_pure_svga", machine_is_svga);
-	retro_set_visibility("dosbox_pure_svgamem", machine_is_svga);
 	retro_set_visibility("dosbox_pure_voodoo", machine_is_svga);
 	retro_set_visibility("dosbox_pure_voodoo_perf", machine_is_svga);
 	retro_set_visibility("dosbox_pure_voodoo_gamma", machine_is_svga);
diff --git a/src/hardware/vga_s3.cpp b/src/hardware/vga_s3.cpp
index 06eafe7..ac82258 100644
--- a/src/hardware/vga_s3.cpp
+++ b/src/hardware/vga_s3.cpp
@@ -546,11 +546,6 @@ void SVGA_Setup_S3Trio(void) {
 	} else if (vga.vmemsize < 4096*1024)	{
 		vga.vmemsize = 3072*1024;
 		vga.s3.reg_36 = 0x5a;		// 3mb fast page mode
-#ifdef C_DBP_LIBRETRO
-	} else if (vga.vmemsize == 8192*1024)	{ // hack more mem than supported by real Trio64 card
-		vga.vmemsize = 8192*1024;
-		vga.s3.reg_36 = 0x7a;		// 8mb fast page mode
-#endif
 	} else {	// Trio64 supported only up to 4M
 		vga.vmemsize = 4096*1024;
 		vga.s3.reg_36 = 0x1a;		// 4mb fast page mode
