--- a/es-core/src/Settings.cpp
+++ b/es-core/src/Settings.cpp
@@ -312,7 +312,7 @@
 	// Audio out device for volume control
 	#ifdef _RPI_
 		mStringMap["AudioDevice"] = "PCM";
-	#elif defined(_ENABLEAMBERELEC) && defined(RG552)
+	#elif defined(_ENABLEAMBERELEC) && (defined(RG552) || defined(RG353))
 		mStringMap["AudioDevice"] = "DAC";
 	#elif defined(_ENABLEAMBERELEC)
 		mStringMap["AudioDevice"] = "Playback";
