diff -rupN /home/fewtarius/repositories/fewtarius/351ELEC/sources/emuelec-emulationstation/emuelec-emulationstation-2afe6efec4e09176882d98323bda5d3f664870a7.orig/es-app/src/guis/GuiMenu.cpp emuelec-emulationstation-2afe6efec4e09176882d98323bda5d3f664870a7/es-app/src/guis/GuiMenu.cpp
--- /home/fewtarius/repositories/fewtarius/351ELEC/sources/emuelec-emulationstation/emuelec-emulationstation-2afe6efec4e09176882d98323bda5d3f664870a7.orig/es-app/src/guis/GuiMenu.cpp	2020-11-12 18:22:26.719272587 -0500
+++ emuelec-emulationstation-2afe6efec4e09176882d98323bda5d3f664870a7/es-app/src/guis/GuiMenu.cpp	2020-11-13 08:53:55.656659931 -0500
@@ -158,7 +158,7 @@ GuiMenu::GuiMenu(Window *window, bool an
 #endif
 
 		addEntry(_("SCRAPE").c_str(), true, [this] { openScraperSettings(); }, "iconScraper");		
-		addEntry(_("UPDATES & DOWNLOADS"), true, [this] { openUpdatesSettings(); }, "iconUpdates");
+		addEntry(_("DOWNLOADS"), true, [this] { openUpdatesSettings(); }, "iconUpdates");
 		addEntry(_("SYSTEM SETTINGS").c_str(), true, [this] { openSystemSettings_batocera(); }, "iconSystem");
 	}
 	else
@@ -1106,7 +1106,7 @@ void GuiMenu::openDeveloperSettings()
 
 void GuiMenu::openUpdatesSettings()
 {
-	GuiSettings *updateGui = new GuiSettings(mWindow, _("UPDATES & DOWNLOADS").c_str());
+	GuiSettings *updateGui = new GuiSettings(mWindow, _("DOWNLOADS").c_str());
 
 	updateGui->addGroup(_("DOWNLOADS"));
 
@@ -1143,49 +1143,6 @@ void GuiMenu::openUpdatesSettings()
 		});
 	}
 
-	updateGui->addGroup(_("SOFTWARE UPDATES"));
-
-	// Enable updates
-	auto updates_enabled = std::make_shared<SwitchComponent>(mWindow);
-	updates_enabled->setState(SystemConf::getInstance()->getBool("updates.enabled"));
-	
-	updateGui->addWithLabel(_("CHECK FOR UPDATES"), updates_enabled);
-	updateGui->addSaveFunc([updates_enabled]
-	{
-		SystemConf::getInstance()->setBool("updates.enabled", updates_enabled->getState());
-	});
-#ifndef _ENABLEEMUELEC
-	auto updatesTypeList = std::make_shared<OptionListComponent<std::string> >(mWindow, _("UPDATE TYPE"), false);
-	
-	std::string updatesType = SystemConf::getInstance()->get("updates.type");
-	if (updatesType.empty() || updatesType != "beta")
-		updatesType = "stable";
-	
-	updatesTypeList->add("stable", "stable", updatesType == "stable");
-	updatesTypeList->add("beta", "beta", updatesType != "stable");
-	
-	updateGui->addWithLabel(_("UPDATE TYPE"), updatesTypeList);
-	updatesTypeList->setSelectedChangedCallback([](std::string name)
-	{
-		if (SystemConf::getInstance()->set("updates.type", name))
-			SystemConf::getInstance()->saveSystemConf();
-	});	
-#endif
-	// Start update
-	updateGui->addEntry(GuiUpdate::state == GuiUpdateState::State::UPDATE_READY ? _("APPLY UPDATE") : _("START UPDATE"), true, [this]
-	{
-		if (GuiUpdate::state == GuiUpdateState::State::UPDATE_READY)
-			quitES(QuitMode::RESTART);
-		else if (GuiUpdate::state == GuiUpdateState::State::UPDATER_RUNNING)
-			mWindow->pushGui(new GuiMsgBox(mWindow, _("UPDATE IS ALREADY RUNNING")));
-		else
-		{
-			if (!checkNetwork())
-				return;
-
-			mWindow->pushGui(new GuiUpdate(mWindow));
-		}
-	});
 	mWindow->pushGui(updateGui);
 }
 
Binary files /home/fewtarius/repositories/fewtarius/351ELEC/sources/emuelec-emulationstation/emuelec-emulationstation-2afe6efec4e09176882d98323bda5d3f664870a7.orig/.git/index and emuelec-emulationstation-2afe6efec4e09176882d98323bda5d3f664870a7/.git/index differ
