diff -rupN emuelec-emulationstation-f89564df948baebf7a0e1f6a6f9ef88557bc84cd.orig/es-app/src/guis/GuiMenu.cpp emuelec-emulationstation-f89564df948baebf7a0e1f6a6f9ef88557bc84cd.new/es-app/src/guis/GuiMenu.cpp
--- emuelec-emulationstation-f89564df948baebf7a0e1f6a6f9ef88557bc84cd.orig/es-app/src/guis/GuiMenu.cpp	2020-11-24 13:29:37.943586866 -0500
+++ emuelec-emulationstation-f89564df948baebf7a0e1f6a6f9ef88557bc84cd.new/es-app/src/guis/GuiMenu.cpp	2020-11-26 10:28:00.879382494 -0500
@@ -3105,6 +3105,13 @@ void GuiMenu::openUISettings()
 
 	s->addGroup(_("DISPLAY OPTIONS"));
 
+        auto notification_time = std::make_shared<SliderComponent>(mWindow, 2.f, 120.f, 2.f, "s");
+        notification_time->setValue(Settings::getInstance()->getInt("notification.display_time"));
+        s->addWithLabel(_("POPUP NOTIFICATION DISPLAY TIMEOUT"), notification_time);
+        s->addSaveFunc([notification_time] {
+                Settings::getInstance()->setInt("notification.display_time", (int)Math::round(notification_time->getValue()));
+        });
+
 	s->addEntry(_("SCREENSAVER SETTINGS"), true, std::bind(&GuiMenu::openScreensaverOptions, this));
 
 	// transition style
@@ -3294,14 +3301,6 @@ void GuiMenu::openSoundSettings()
 		Settings::getInstance()->setBool("audio.display_titles", display_titles->getState());
 	});
 
-	// batocera - how long to display the song titles?
-	auto titles_time = std::make_shared<SliderComponent>(mWindow, 2.f, 120.f, 2.f, "s");
-	titles_time->setValue(Settings::getInstance()->getInt("audio.display_titles_time"));
-	s->addWithLabel(_("HOW MANY SECONDS FOR SONG TITLES"), titles_time);
-	s->addSaveFunc([titles_time] {
-		Settings::getInstance()->setInt("audio.display_titles_time", (int)Math::round(titles_time->getValue()));
-	});
-
 	// batocera - music per system
 	auto music_per_system = std::make_shared<SwitchComponent>(mWindow);
 	music_per_system->setState(Settings::getInstance()->getBool("audio.persystem"));
diff -rupN emuelec-emulationstation-f89564df948baebf7a0e1f6a6f9ef88557bc84cd.orig/es-core/src/Settings.cpp emuelec-emulationstation-f89564df948baebf7a0e1f6a6f9ef88557bc84cd.new/es-core/src/Settings.cpp
--- emuelec-emulationstation-f89564df948baebf7a0e1f6a6f9ef88557bc84cd.orig/es-core/src/Settings.cpp	2020-11-24 13:29:37.951587402 -0500
+++ emuelec-emulationstation-f89564df948baebf7a0e1f6a6f9ef88557bc84cd.new/es-core/src/Settings.cpp	2020-11-26 09:49:58.366924854 -0500
@@ -290,10 +290,12 @@ void Settings::setDefaults()
 	mBoolMap["audio.persystem"] = false;
 	mBoolMap["audio.display_titles"] = true;
 	mBoolMap["audio.thememusics"] = true;	
-	mIntMap["audio.display_titles_time"] = 10;
 	
 	mBoolMap["NetPlayCheckIndexesAtStart"] = false;
 
+	// Default notification display time
+	mIntMap["notification.display_time"] = 4;
+
 #if WIN32
 	mBoolMap["updates.enabled"] = true;
 	mBoolMap["global.retroachievements"] = false;
diff -rupN emuelec-emulationstation-f89564df948baebf7a0e1f6a6f9ef88557bc84cd.orig/es-core/src/Window.cpp emuelec-emulationstation-f89564df948baebf7a0e1f6a6f9ef88557bc84cd.new/es-core/src/Window.cpp
--- emuelec-emulationstation-f89564df948baebf7a0e1f6a6f9ef88557bc84cd.orig/es-core/src/Window.cpp	2020-11-24 13:29:37.951587402 -0500
+++ emuelec-emulationstation-f89564df948baebf7a0e1f6a6f9ef88557bc84cd.new/es-core/src/Window.cpp	2020-11-26 09:49:26.227228064 -0500
@@ -275,9 +275,9 @@ void Window::displayNotificationMessage(
 
 	if (duration <= 0)
 	{
-		duration = Settings::getInstance()->getInt("audio.display_titles_time");
+		duration = Settings::getInstance()->getInt("notification.display_time");
 		if (duration <= 2 || duration > 120)
-			duration = 10;
+			duration = 4;
 
 		duration *= 1000;
 	}
