diff --git a/src/audio.c b/src/audio.c
index 26877e9..c91427b 100644
--- a/src/audio.c
+++ b/src/audio.c
@@ -173,7 +173,7 @@ uint32_t go2_audio_volume_get(go2_audio_t* audio)
     snd_mixer_t *handle;
     snd_mixer_selem_id_t *sid;
     const char *card = "default";
-    const char *selem_name = "Playback";
+    const char *selem_name = "DAC";
 
     snd_mixer_open(&handle, 0);
     snd_mixer_attach(handle, card);
@@ -210,7 +210,7 @@ void go2_audio_volume_set(go2_audio_t* audio, uint32_t value)
     snd_mixer_t *handle;
     snd_mixer_selem_id_t *sid;
     const char *card = "default";
-    const char *selem_name = "Playback";
+    const char *selem_name = "DAC";
 
     snd_mixer_open(&handle, 0);
     snd_mixer_attach(handle, card);
