diff --git a/src/javax/microedition/rms/RecordStore.java b/src/javax/microedition/rms/RecordStore.java
index 802d3d1..fa4b87d 100644
--- a/src/javax/microedition/rms/RecordStore.java
+++ b/src/javax/microedition/rms/RecordStore.java
@@ -70,8 +70,8 @@ public class RecordStore
 
 		appname = Mobile.getPlatform().loader.suitename;
 
-		rmsPath = Mobile.getPlatform().dataPath + "./rms/"+appname;
-		rmsFile = Mobile.getPlatform().dataPath + "./rms/"+appname+"/"+recordStoreName;
+		rmsPath = "/roms/j2me/rms/"+appname;
+		rmsFile = "/roms/j2me/rms/"+appname+"/"+recordStoreName;
 
 		try
 		{
@@ -244,7 +244,7 @@ public class RecordStore
 	{
 		try
 		{
-			File fstore = new File(Mobile.getPlatform().dataPath + "./rms/"+Mobile.getPlatform().loader.suitename+"/"+recordStoreName);
+			File fstore = new File("/roms/j2me/rms/"+Mobile.getPlatform().loader.suitename+"/"+recordStoreName);
 			fstore.delete();
 		}
 		catch (Exception e)
@@ -347,7 +347,7 @@ public class RecordStore
 		//System.out.println("List Record Stores");
 		if(rmsPath==null)
 		{
-			rmsPath = Mobile.getPlatform().dataPath + "./rms/"+Mobile.getPlatform().loader.name;
+			rmsPath = "/roms/j2me/rms/"+Mobile.getPlatform().loader.name;
 			try
 			{
 				Files.createDirectories(Paths.get(rmsPath));
