diff --git a/src/video/SDL_video.c b/src/video/SDL_video.c
index 059316a5d..b344117bb 100644
--- a/src/video/SDL_video.c
+++ b/src/video/SDL_video.c
@@ -1443,6 +1443,7 @@ SDL_ToggleDragAndDropSupport(void)
 static void
 SDL_FinishWindowCreation(SDL_Window *window, Uint32 flags)
 {
+	SDL_Mouse *mouse = SDL_GetMouse();
     PrepareDragAndDropSupport(window);
 
     if (flags & SDL_WINDOW_MAXIMIZED) {
@@ -1461,6 +1462,10 @@ SDL_FinishWindowCreation(SDL_Window *window, Uint32 flags)
            the keyboard grabbed too and SDL_SetWindowMouseGrab() won't do that.
         */
         SDL_SetWindowGrab(window, SDL_TRUE);
+        
+        mouse->x = window->x+1;
+        mouse->y = window->y+1;
+        
     }
     if (flags & SDL_WINDOW_KEYBOARD_GRABBED) {
         SDL_SetWindowKeyboardGrab(window, SDL_TRUE);
