diff -rupN u-boot-e7b255b54c42c8a805dee7a9409a8838cfa13586.orig/board/rockchip/odroidgo2/odroidgo2.c u-boot-e7b255b54c42c8a805dee7a9409a8838cfa13586/board/rockchip/odroidgo2/odroidgo2.c
--- a/board/rockchip/odroidgoa/odroidgoa.c	2020-11-21 06:20:02.807164746 -0500
+++ b/board/rockchip/odroidgoa/odroidgoa.c	2020-11-21 06:21:16.512742672 -0500
@@ -132,10 +132,10 @@ void board_check_mandatory_files(void)
 	}
 
 	/* check launcher in ext4 fs of sd card */
-	if (file_exists("mmc", "1:2", "/usr/local/bin/emulationstation/emulationstation",
+	if (!file_exists("mmc", "1:2", "/usr/local/bin/emulationstation/emulationstation",
 				FS_TYPE_EXT)) {
 		lcd_setfg_color("white");
-		lcd_printf(0, 0, 1, "[ GO Advanced EMULATION Image ]");
+		lcd_printf(0, 0, 1, "[ 351ELEC - Anbernic RG351P ]");
 	}
 
 	return;
