diff --git a/Makefile b/Makefile
index 58648e5..a1061dc 100644
--- a/Makefile
+++ b/Makefile
@@ -356,34 +356,26 @@ else ifneq (,$(findstring arm64_cortex_a53_gles3,$(platform)))
    COREFLAGS += -DOS_LINUX
    ASFLAGS = -f elf64 -d ELF_TYPE
 
-# Rockchip RK3288 e.g. Asus Tinker Board / RK3328 e.g. PINE64 Rock64 / RK3399 e.g. PINE64 RockPro64 - 32-bit userspace
+# Rockchip RK3326,RK3328,RK3399
 else ifneq (,$(findstring RK,$(platform)))
    TARGET := $(TARGET_NAME)_libretro.so
    LDFLAGS += -shared -Wl,--version-script=$(LIBRETRO_DIR)/link.T -Wl,--no-undefined -ldl
 
    ifneq (,$(findstring RK33,$(platform)))
-      CPUFLAGS += -march=armv8-a+crc -mfloat-abi=hard -mfpu=neon-fp-armv8
       ifneq (,$(findstring RK3399,$(platform)))
-         CPUFLAGS += -mtune=cortex-a72.cortex-a53
-         GLES3 = 1
+         CPUFLAGS += -march=armv8-a -mtune=cortex-a72.cortex-a53
       else ifneq (,$(findstring RK3328,$(platform)))
          CPUFLAGS += -mtune=cortex-a53
-         GLES = 1
+      else ifneq (,$(findstring RK3326,$(platform)))
+         CPUFLAGS += -mtune=cortex-a35
       endif
-   else ifneq (,$(findstring RK3288,$(platform)))
-      CPUFLAGS += -march=armv7ve -mtune=cortex-a17 -mfloat-abi=hard -mfpu=neon-vfpv4
-      GLES3 = 1
-   endif
-
-   ifneq (,$(findstring mesa,$(platform)))
-      COREFLAGS += -DEGL_NO_X11
    endif
 
-   GL_LIB := -lGLESv2
-   HAVE_NEON = 1
-   WITH_DYNAREC=arm
-   COREFLAGS += -DUSE_GENERIC_GLESV2 -DOS_LINUX
-   ASFLAGS = -f elf -d ELF_TYPE
+   GLES3 = 1
+   GL_LIB := -lGLESv3
+   WITH_DYNAREC = aarch64
+   COREFLAGS += -DOS_LINUX
+   ASFLAGS = -f elf64 -d ELF_TYPE
 
 # OS X
 else ifneq (,$(findstring osx,$(platform)))
