diff -rupN emuelec-emulationstation-2afe6efec4e09176882d98323bda5d3f664870a7.orig/es-app/src/guis/GuiMenu.cpp emuelec-emulationstation-2afe6efec4e09176882d98323bda5d3f664870a7/es-app/src/guis/GuiMenu.cpp
--- emuelec-emulationstation-2afe6efec4e09176882d98323bda5d3f664870a7.orig/es-app/src/guis/GuiMenu.cpp	2020-11-12 18:22:26.719272587 -0500
+++ emuelec-emulationstation-2afe6efec4e09176882d98323bda5d3f664870a7/es-app/src/guis/GuiMenu.cpp	2020-11-12 19:54:05.473588305 -0500
@@ -796,26 +796,18 @@ void GuiMenu::openSystemInformations_bat
 
 	Window *window = mWindow;
 	bool isFullUI = UIModeController::getInstance()->isUIModeFull();
-	GuiSettings *informationsGui = new GuiSettings(window, _("INFORMATION").c_str());
+	GuiSettings *informationsGui = new GuiSettings(window, _("SYSTEM INFORMATION").c_str());
 
 	auto version = std::make_shared<TextComponent>(window, ApiSystem::getInstance()->getVersion(), font, color);
 	informationsGui->addWithLabel(_("VERSION"), version);
 
 	bool warning = ApiSystem::getInstance()->isFreeSpaceLimit();
 	auto userspace = std::make_shared<TextComponent>(window,
-		ApiSystem::getInstance()->getFreeSpaceUserInfo(),
+		ApiSystem::getInstance()->getFreeSpaceInfo("/storage/roms"),
 		font,
 		warning ? 0xFF0000FF : color);
 	informationsGui->addWithLabel(_("USER DISK USAGE"), userspace);
 
-#ifndef _ENABLEEMUELEC
-	auto systemspace = std::make_shared<TextComponent>(window,
-		ApiSystem::getInstance()->getFreeSpaceSystemInfo(),
-		font,
-		color);
-	informationsGui->addWithLabel(_("SYSTEM DISK USAGE"), systemspace);
-#endif
-
 	// various informations
 	std::vector<std::string> infos = ApiSystem::getInstance()->getSystemInformations();
 	for (auto it = infos.begin(); it != infos.end(); it++) {
Binary files emuelec-emulationstation-2afe6efec4e09176882d98323bda5d3f664870a7.orig/.git/index and emuelec-emulationstation-2afe6efec4e09176882d98323bda5d3f664870a7/.git/index differ
