diff --git a/UI/NativeApp.cpp b/UI/NativeApp.cpp
index 7c26bb348e..8233e973a4 100644
--- a/UI/NativeApp.cpp
+++ b/UI/NativeApp.cpp
@@ -1529,6 +1529,9 @@ bool Native_UpdateScreenScale(int width, int height) {
 	float g_logical_dpi = System_GetPropertyFloat(SYSPROP_DISPLAY_LOGICAL_DPI);
 	g_display.dpi = System_GetPropertyFloat(SYSPROP_DISPLAY_DPI);
 
+	// override display dpi
+	g_display.dpi = 72.0f;
+
 	if (g_display.dpi < 0.0f) {
 		g_display.dpi = 96.0f;
 	}
