diff --git a/common/libretro.c b/common/libretro.c
index 4f48b90..db07276 100644
--- a/common/libretro.c
+++ b/common/libretro.c
@@ -270,7 +270,12 @@ void retro_set_rumble_damage(int damage)
    if (strength == rumble_damage_strength)
       return;
 
-   rumble.set_rumble_state(0, RETRO_RUMBLE_STRONG, strength);
+   uint pwm_duty_cycle = 1000000 - (int)(((double)strength / 65535.0) * 1000000.0);
+   FILE *fp;
+   fp = fopen("/sys/class/pwm/pwmchip0/pwm0/duty_cycle", "w");
+   fprintf(fp, "%u", pwm_duty_cycle);
+   fclose(fp);
+
    rumble_damage_strength = strength;
 }
 
@@ -296,7 +301,12 @@ void retro_set_rumble_touch(unsigned intensity, float duration)
    if (strength == rumble_touch_strength)
       return;
 
-   rumble.set_rumble_state(0, RETRO_RUMBLE_WEAK, strength);
+   uint pwm_duty_cycle = 1000000 - (int)(((double)strength / 65535.0) * 1000000.0);
+   FILE *fp;
+   fp = fopen("/sys/class/pwm/pwmchip0/pwm0/duty_cycle", "w");
+   fprintf(fp, "%u", pwm_duty_cycle);
+   fclose(fp);
+   
    rumble_touch_strength = strength;
 }
 
