diff -rupN PPSSPPSDL-087de849bdc74205dd00d8e6e11ba17a591213ab.orig/SDL/SDLMain.cpp PPSSPPSDL-087de849bdc74205dd00d8e6e11ba17a591213ab/SDL/SDLMain.cpp
--- PPSSPPSDL-087de849bdc74205dd00d8e6e11ba17a591213ab.orig/SDL/SDLMain.cpp	2020-11-27 21:15:14.705925357 -0500
+++ PPSSPPSDL-087de849bdc74205dd00d8e6e11ba17a591213ab/SDL/SDLMain.cpp	2020-11-27 21:48:43.295100516 -0500
@@ -586,8 +586,8 @@ int main(int argc, char *argv[]) {
 		g_Config.bFullScreen = true;
 	} else {
 		// set a sensible default resolution (2x)
-		pixel_xres = 480 * 2 * set_scale;
-		pixel_yres = 272 * 2 * set_scale;
+		pixel_xres = 320;
+		pixel_yres = 240;
 		if (portrait) {
 			std::swap(pixel_xres, pixel_yres);
 		}
