diff -rupN mgba-43bc47742f5c642559d259d461cd16278f89e604.orig/src/platform/libretro/libretro.c mgba-43bc47742f5c642559d259d461cd16278f89e604/src/platform/libretro/libretro.c
--- mgba-43bc47742f5c642559d259d461cd16278f89e604.orig/src/platform/libretro/libretro.c	2020-11-18 04:33:56.000000000 -0500
+++ mgba-43bc47742f5c642559d259d461cd16278f89e604/src/platform/libretro/libretro.c	2020-11-18 07:46:36.015486575 -0500
@@ -349,15 +349,17 @@ void retro_run(void) {
 	videoCallback(outputBuffer, width, height, BYTES_PER_PIXEL * 256);
 
 	if (rumbleCallback) {
-		if (rumbleUp) {
-			rumbleCallback(0, RETRO_RUMBLE_STRONG, rumbleUp * 0xFFFF / (rumbleUp + rumbleDown));
-			rumbleCallback(0, RETRO_RUMBLE_WEAK, rumbleUp * 0xFFFF / (rumbleUp + rumbleDown));
-		} else {
-			rumbleCallback(0, RETRO_RUMBLE_STRONG, 0);
-			rumbleCallback(0, RETRO_RUMBLE_WEAK, 0);
-		}
-		rumbleUp = 0;
-		rumbleDown = 0;
+            if (rumbleUp) {
+                system("echo 10 > /sys/class/pwm/pwmchip0/pwm0/duty_cycle");
+                rumbleCallback(0, RETRO_RUMBLE_STRONG, rumbleUp * 0xFFFF / (rumbleUp + rumbleDown));
+                rumbleCallback(0, RETRO_RUMBLE_WEAK, rumbleUp * 0xFFFF / (rumbleUp + rumbleDown));
+            } else {
+                system("echo 1000000 > /sys/class/pwm/pwmchip0/pwm0/duty_cycle");
+                rumbleCallback(0, RETRO_RUMBLE_STRONG, 0);
+                rumbleCallback(0, RETRO_RUMBLE_WEAK, 0);
+            }
+            rumbleUp = 0;
+            rumbleDown = 0;
 	}
 }
 
