diff --git a/cmake/FindCapsImage.cmake b/cmake/FindCapsImage.cmake
index 982e5285..94ada2fe 100644
--- a/cmake/FindCapsImage.cmake
+++ b/cmake/FindCapsImage.cmake
@@ -6,7 +6,7 @@ ENDIF (CAPSIMAGE_INCLUDE_DIR)
 
 
 # Choose the library version to use : 4 or 5
-SET(CAPSIMAGE_VERSION 4)
+SET(CAPSIMAGE_VERSION 5)
 
 
 if(CAPSIMAGE_VERSION STREQUAL 4)
diff --git a/src/floppy_ipf.c b/src/floppy_ipf.c
index c615b755..8e52b712 100644
--- a/src/floppy_ipf.c
+++ b/src/floppy_ipf.c
@@ -24,10 +24,16 @@ const char floppy_ipf_fileid[] = "Hatari floppy_ipf.c : " __DATE__ " " __TIME__;
 #include "screen.h"
 #include "video.h"
 #include "cycles.h"
+#include <stdint.h>
 
 #ifdef HAVE_CAPSIMAGE
+#ifndef __cdecl
+#define __cdecl  /* CAPS headers need this, but do not define it on their own */
+#endif
 #if CAPSIMAGE_VERSION == 5
-#include <CapsLibAll.h>
+#include <caps5/CapsLibAll.h>
+#define CapsLong SDWORD
+#define CapsULong UDWORD
 #else
 #include <caps/fdc.h>
 #define CAPS_LIB_RELEASE	4
diff --git a/src/uae-cpu/newcpu.c b/src/uae-cpu/newcpu.c
index e752e83d..91cf41cf 100644
--- a/src/uae-cpu/newcpu.c
+++ b/src/uae-cpu/newcpu.c
@@ -162,6 +162,9 @@ const char NewCpu_fileid[] = "Hatari newcpu.c : " __DATE__ " " __TIME__;
 #include "68kDisass.h"
 
 #ifdef HAVE_CAPSIMAGE
+#ifndef __cdecl
+#define __cdecl  /* CAPS headers need this, but do not define it on their own */
+#endif
 #if CAPSIMAGE_VERSION == 5
-#include <CapsLibAll.h>
+#include <caps5/CapsLibAll.h>
 #endif
