--- a/src/def.h
+++ b/src/def.h
@@ -43,7 +43,7 @@
    #define KEYBOARD_SYMBOL_SIZE     24

 // Parameters for RG351MP
-#elif defined(DEVICE_RG351MP)
+#elif defined(DEVICE_RG351MP) || defined(DEVICE_RG353)
    #define SCREEN_WIDTH             640
    #define SCREEN_HEIGHT            480
    #define HARDWARE_ACCELERATION    0
@@ -188,7 +188,7 @@
    #define BUTTON_HELD_PAGEDOWN           SDL_JoystickGetButton(g_joystick, 5) || SDL_JoystickGetButton(g_joystick, 11)
    #define BUTTON_HELD_SELECT             SDL_JoystickGetButton(g_joystick, 3)
    #define BUTTON_HELD_VALIDATE           SDL_JoystickGetButton(g_joystick, 0)
-#elif defined(DEVICE_RG351MP)
+#elif defined(DEVICE_RG351MP) || defined(DEVICE_RG353)
    #define BUTTON_PRESSED_UP              event.type == SDL_JOYBUTTONDOWN && event.jbutton.button == 8
    #define BUTTON_PRESSED_DOWN            event.type == SDL_JOYBUTTONDOWN && event.jbutton.button == 9
    #define BUTTON_PRESSED_LEFT            event.type == SDL_JOYBUTTONDOWN && event.jbutton.button == 10
