diff -rupN eduke.orig/Common.mak eduke.new/Common.mak
--- eduke.orig/Common.mak	2021-02-02 17:20:26.529600337 -0500
+++ eduke.new/Common.mak	2021-02-02 17:20:48.166006894 -0500
@@ -986,9 +986,6 @@ else ifeq ($(PLATFORM),WII)
     LIBS += -laesnd_tueidj -lfat -lwiiuse -lbte -lwiikeyboard -logc
 else ifeq ($(SUBPLATFORM),LINUX)
     LIBS += -lrt
-    ifneq ($(findstring armv6,$(IMPLICIT_ARCH))$(findstring armv7,$(IMPLICIT_ARCH))$(findstring aarch64,$(IMPLICIT_ARCH)),)
-        LIBS += -latomic
-    endif
 endif
 
 ifeq (,$(filter $(PLATFORM),WINDOWS WII))
