diff --git a/libretro/libretro.c b/libretro/libretro.c
index f68d93f..63efdcf 100644
--- a/libretro/libretro.c
+++ b/libretro/libretro.c
@@ -385,7 +385,7 @@ void retro_get_system_av_info(struct retro_system_av_info *info)
   info->geometry.base_height = SCREENHEIGHT;
   info->geometry.max_width = SCREENWIDTH;
   info->geometry.max_height = SCREENHEIGHT;
-  info->geometry.aspect_ratio = 4.0 / 3.0;
+  info->geometry.aspect_ratio = 3.0 / 2.0;
 }


diff --git a/libretro/libretro_core_options.h b/libretro/libretro_core_options.h
index e764703..dc65f4a 100644
--- a/libretro/libretro_core_options.h
+++ b/libretro/libretro_core_options.h
@@ -33,6 +33,7 @@ struct retro_core_option_definition option_defs_us[] = {
       "Configure the resolution. Requires a restart.",
       {
          { "320x200",   NULL },
+         { "480x320",   NULL },
          { "640x400",   NULL },
          { "960x600",   NULL },
          { "1280x800",  NULL },
@@ -42,7 +43,7 @@ struct retro_core_option_definition option_defs_us[] = {
          { "2560x1600", NULL },
          { NULL, NULL },
       },
-      "320x200"
+      "480x320"
    },
    {
       "prboom-mouse_on",
