diff -rupN devilutionX.orig/SourceX/DiabloUI/selgame.cpp devilutionX.new/SourceX/DiabloUI/selgame.cpp
--- devilutionX.orig/SourceX/DiabloUI/selgame.cpp	2021-01-14 07:40:13.640658248 -0500
+++ devilutionX.new/SourceX/DiabloUI/selgame.cpp	2021-01-14 07:50:54.667214440 -0500
@@ -70,6 +70,7 @@ void selgame_GameSelection_Init()
 	}
 
 	getIniValue("Phone Book", "Entry1", selgame_Ip, 128);
+        getIniValue("Phone Book", "Password1", selgame_Password, 15);
 
 	selgame_FreeVectors();
 
@@ -350,7 +351,7 @@ void selgame_Speed_Select(int value)
 
 void selgame_Password_Init(int value)
 {
-	memset(&selgame_Password, 0, sizeof(selgame_Password));
+	//memset(&selgame_Password, 0, sizeof(selgame_Password));
 
 	selgame_FreeVectors();
 
