diff -rupN apitrace.orig/dispatch/glimports.hpp apitrace.new/dispatch/glimports.hpp
--- apitrace.orig/dispatch/glimports.hpp	2021-01-08 19:52:31.768258806 -0500
+++ apitrace.new/dispatch/glimports.hpp	2021-01-08 19:55:45.511116819 -0500
@@ -37,7 +37,7 @@
 #  include <windows.h>
 #elif defined(__APPLE__)
 #elif defined(HAVE_X11)
-#  include <X11/Xlib.h>
+//#  include <X11/Xlib.h>
 #endif /* !_WIN32 */
 
 
diff -rupN apitrace.orig/thirdparty/khronos/EGL/eglplatform.h apitrace.new/thirdparty/khronos/EGL/eglplatform.h
--- apitrace.orig/thirdparty/khronos/EGL/eglplatform.h	2021-01-08 19:52:31.792259353 -0500
+++ apitrace.new/thirdparty/khronos/EGL/eglplatform.h	2021-01-08 19:58:22.843613539 -0500
@@ -125,8 +125,8 @@ typedef khronos_uintptr_t EGLNativeWindo
 #elif defined(__unix__) || defined(USE_X11)
 
 /* X11 (tentative)  */
-#include <X11/Xlib.h>
-#include <X11/Xutil.h>
+//#include <X11/Xlib.h>
+//#include <X11/Xutil.h>
 
 typedef Display *EGLNativeDisplayType;
 typedef Pixmap   EGLNativePixmapType;
diff -rupN apitrace.orig/thirdparty/khronos/GL/glx.h apitrace.new/thirdparty/khronos/GL/glx.h
--- apitrace.orig/thirdparty/khronos/GL/glx.h	2021-01-08 19:52:31.796259444 -0500
+++ apitrace.new/thirdparty/khronos/GL/glx.h	2021-01-08 19:53:57.757888475 -0500
@@ -35,8 +35,8 @@
 ** compliant with the OpenGL(R) version 1.2.1 Specification.
 */
 
-#include <X11/Xlib.h>
-#include <X11/Xutil.h>
+//#include <X11/Xlib.h>
+//#include <X11/Xutil.h>
 #include <GL/gl.h>
 
 #ifdef __cplusplus
