From cdcf4795b53caaafe2bb4194fdc504bd848fa42c Mon Sep 17 00:00:00 2001
From: YonKuma <kumarei@gmail.com>
Date: Sat, 23 Oct 2021 20:56:35 -0400
Subject: [PATCH] Fix for V Deadzone

---
 drivers/input/evdev.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/input/evdev.c b/drivers/input/evdev.c
index 632e3797d98f..ef7affb862b5 100644
--- a/drivers/input/evdev.c
+++ b/drivers/input/evdev.c
@@ -15,14 +15,14 @@
 #define EVDEV_MIN_BUFFER_SIZE	64U
 #define EVDEV_BUF_PACKETS	8
 
-#define RG351P_ABS_Z_MIN 500
+#define RG351P_ABS_Z_MIN 700
 #define RG351P_ABS_Z_MAX 3500
 #define RG351P_ABS_RY_MIN 500
-#define RG351P_ABS_RY_MAX 3500
-#define RG351P_ABS_RX_MIN 500
-#define RG351P_ABS_RX_MAX 3200
+#define RG351P_ABS_RY_MAX 3200
+#define RG351P_ABS_RX_MIN 700
+#define RG351P_ABS_RX_MAX 3500
 #define RG351P_ABS_RZ_MIN 500
-#define RG351P_ABS_RZ_MAX 3200
+#define RG351P_ABS_RZ_MAX 3500
 
 #include <linux/poll.h>
 #include <linux/sched.h>
-- 
2.25.1

