diff -rupN flycast-0e10e86ea9ca0f8655c98909da7a845e7643b36f.orig/core/libretro/libretro.cpp flycast-0e10e86ea9ca0f8655c98909da7a845e7643b36f/core/libretro/libretro.cpp
--- flycast-0e10e86ea9ca0f8655c98909da7a845e7643b36f.orig/core/libretro/libretro.cpp	2020-11-17 06:19:24.446176432 -0500
+++ flycast-0e10e86ea9ca0f8655c98909da7a845e7643b36f/core/libretro/libretro.cpp	2020-12-01 20:12:30.163141298 -0500
@@ -3111,10 +3111,25 @@ void UpdateVibration(u32 port, u32 value
    u8 POW_NEG = (value >> 12) & 0x7;
    u8 FREQ = (value >> 16) & 0xFF;
    s16 INC = (value >> 24) & 0xFF;
-   if (value & 0x8000)			// INH
-	  INC = -INC;
-   else if (!(value & 0x0800))	// EXH
-	  INC = 0;
+   //if (value & 0x8000)			// INH
+   //	  INC = -INC;
+   //else if (!(value & 0x0800))	// EXH
+   //	  INC = 0;
+
+   // Rumble for the RG351P
+   FILE *fp;
+   if (value <= 1000000) {
+	fp = fopen("/sys/class/pwm/pwmchip0/pwm0/duty_cycle", "w");
+	fprintf(fp, "1000000");
+	fclose(fp);
+	INC = -INC;
+   } else {
+	fp = fopen("/sys/class/pwm/pwmchip0/pwm0/duty_cycle", "w");
+	fprintf(fp, "10");
+	fclose(fp);
+	INC = 0;
+   }
+
    bool CNT = value & 1;
 
    double pow = min((POW_POS + POW_NEG) / 7.0, 1.0);
