diff -rupN flycast-0e10e86ea9ca0f8655c98909da7a845e7643b36f.orig/core/libretro/libretro.cpp flycast-0e10e86ea9ca0f8655c98909da7a845e7643b36f/core/libretro/libretro.cpp
--- flycast-0e10e86ea9ca0f8655c98909da7a845e7643b36f.orig/core/libretro/libretro.cpp	2020-11-17 06:19:24.446176432 -0500
+++ flycast-0e10e86ea9ca0f8655c98909da7a845e7643b36f/core/libretro/libretro.cpp	2020-11-17 06:21:21.142725535 -0500
@@ -3111,10 +3111,12 @@ void UpdateVibration(u32 port, u32 value
    u8 POW_NEG = (value >> 12) & 0x7;
    u8 FREQ = (value >> 16) & 0xFF;
    s16 INC = (value >> 24) & 0xFF;
-   if (value & 0x8000)			// INH
-	  INC = -INC;
-   else if (!(value & 0x0800))	// EXH
-	  INC = 0;
+   if (value <= 1000000){            // INH
+      system("echo 1000000 > /sys/class/pwm/pwmchip0/pwm0/duty_cycle");
+      INC = -INC;}
+   else{    // EXH
+        system("echo 10 > /sys/class/pwm/pwmchip0/pwm0/duty_cycle && sleep 0.4 && echo 1000000 > /sys/class/pwm/pwmchip0/pwm0/duty_cycle &");
+      INC = 0;}
    bool CNT = value & 1;
 
    double pow = min((POW_POS + POW_NEG) / 7.0, 1.0);
@@ -3123,13 +3125,13 @@ void UpdateVibration(u32 port, u32 value
    rumble.set_rumble_state(port, RETRO_RUMBLE_STRONG, (u16)(65535 * pow));
 
    if (FREQ > 0 && (!CNT || INC))
-	  vib_stop_time[port] = get_time_ms() + min((int)(1000 * (INC ? abs(INC) * max(POW_POS, POW_NEG) : 1) / FREQ), (int)max_duration);
+      vib_stop_time[port] = get_time_ms() + min((int)(1000 * (INC ? abs(INC) * max(POW_POS, POW_NEG) : 1) / FREQ), (int)max_duration);
    else
-	  vib_stop_time[port] = get_time_ms() + max_duration;
+      vib_stop_time[port] = get_time_ms() + max_duration;
    if (INC == 0 || pow == 0)
-	  vib_delta[port] = 0.0;
+      vib_delta[port] = 0.0;
    else
-	  vib_delta[port] = FREQ / (1000.0 * INC * max(POW_POS, POW_NEG));
+      vib_delta[port] = FREQ / (1000.0 * INC * max(POW_POS, POW_NEG));
 }
 
 extern u8 kb_shift; 		// shift keys pressed (bitmask)
