diff --git a/src/floppybridge/CommonBridgeTemplate.h b/src/floppybridge/CommonBridgeTemplate.h
index e70bbede..8419d7d7 100644
--- a/src/floppybridge/CommonBridgeTemplate.h
+++ b/src/floppybridge/CommonBridgeTemplate.h
@@ -35,6 +35,7 @@
 #include <vector>
 #include <deque>
 #include <mutex>
+#include <string>
 #include <condition_variable>
 #include "floppybridge_abstract.h"
 #include "RotationExtractor.h"
diff --git a/src/floppybridge/SerialIO.h b/src/floppybridge/SerialIO.h
index abb43858..a73daf7e 100644
--- a/src/floppybridge/SerialIO.h
+++ b/src/floppybridge/SerialIO.h
@@ -26,6 +26,7 @@
 
 #include <string>
 #include <vector>
+#include <stdint.h>
 
 #ifdef _WIN32
 #ifdef USING_MFC
